---
title: Xenium
schema_name: xenium
category: Spatial Transcriptomics
all_versions_deprecated: False
exclude_from_index: False
layout: default

---
Prepare your metadata based on the latest metadata schema using one of the template files below. See the instructions in the [Metadata Validation Workflow](https://docs.google.com/document/d/1lfgiDGbyO4K4Hz1FMsJjmJd9RdwjShtJqFYNwKpbcZY) document for more information on preparing and validating your metadata.tsv file prior to submission.

Related files:


- [üìù Excel template](https://raw.githubusercontent.com/hubmapconsortium/dataset-metadata-spreadsheet/main/xenium/latest/xenium.xlsx): For metadata entry.
- [üìù TSV template](https://raw.githubusercontent.com/hubmapconsortium/dataset-metadata-spreadsheet/main/xenium/latest/xenium.tsv): Alternative for metadata entry.




## Metadata schema


<summary><a href="https://openview.metadatacenter.org/templates/https:%2F%2Frepo.metadatacenter.org%2Ftemplates%2Fc83b9743-ea76-4eb3-97f1-7325331936ea"><b>Version 2 (use this one)</b></a></summary>



<br>

## Directory schemas
<summary><b>Version 3.8 (use this one)</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/markers\.csv$</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection. This should minimally contain name of fluorophor, vendor, channel. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/custom_probe_set\.bed$</code> |  | This is a BED file version of the custom probe set file. |
| <code>raw\/transcript_locations\.csv$</code> | ‚úì | Contains decription of the location of all decoded transcripts. The origin of the coordinate is 0,0 at the top left corner of the image. The file should include: gene name, x, y, z (optional), quality score (optional). It is expected that the first row in the file contains the column header. |
| <code>raw\/custom_gene_list\.csv$</code> |  | This describes the target genes profiled by the assay. For advanced design, this can be probes sequences for splicing or other analysis for any target of interest. The format should minimally contain: gene name, ensemble ID |
| <code>raw\/probes\.csv$</code> |  | A CSV file describing the probe panel used. This is tyipcally what's used to specific the probe set when ordering a probe panel for a Xenium run. |
| <code>raw\/gene_panel\.json$</code> | ‚úì | This is the JSON file describing the probes, as output from the xenium-ranger pipeline. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tif&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*tissue-boundary\.geojson$</code> |  | **[QA/QC]** If the boundaries of the tissue have been identified (e.g., by manual efforts), then the boundary geometry can be included as a GeoJSON file named ‚Äú*.tissue-boundary.geojson‚Äù. |
| <code>lab_processed\/xenium_bundle\/.*</code> | ‚úì | The Xenium bundle includes *.zarr, *.parquet, *.tiff and *.gz files et al. Zarr files are required to view and explore data in Xenium Explorer. Parquet files are faster than CSV files and used extensively by 10X Genomics in Xenium and Visium HD. Sites are free to include in this subdirectory as many Xenium bundle files as possible: <https://www.10xgenomics.com/support/software/xenium-onboard-analysis/latest/analysis/xoa-output-understanding-outputs> |
| <code>lab_processed\/xenium_bundle\/cell_feature_matrix\.h5$</code> | ‚úì | Contains cell feature matrix |
| <code>lab_processed\/xenium_bundle\/experiment\.xenium$</code> | ‚úì | Contains specifications |
| <code>lab_processed\/xenium_bundle\/nucleus_boundaries\.parquet$</code> | ‚úì | Contains polygons of nucleus boundaries |
| <code>lab_processed\/xenium_bundle\/cell_boundaries\.parquet$</code> | ‚úì | Contains polygons of cell boundaries |
| <code>lab_processed\/xenium_bundle\/transcripts\.parquet$</code> | ‚úì | Contains transcripts |
| <code>lab_processed\/xenium_bundle\/cells\.parquet$</code> | ‚úì | Contains cell metadata |
| <code>lab_processed\/xenium_bundle\/morphology_mip\.ome\.tif$</code> |  | Contains morphology mip |
| <code>lab_processed\/xenium_bundle\/morphology_focus\.ome\.tif$</code> |  | Morphology focus file for Xenium datasets with a single morphology focus file |
| <code>lab_processed\/xenium_bundle\/transcripts\.zarr\.zip$</code> |  | The transcripts file made by the Xenium onboard analysis pipeline containing transcript quality and localization based on the image coordinate system of the morphology images. |
| <code>lab_processed\/xenium_bundle\/cells\.zarr\.zip$</code> |  | Segmentation masks and boundaries for nuclei and cells made by the Xenium onboard analysis pipeline, used for assigning transcripts to cells. |
| <code>lab_processed\/xenium_bundle\/cell_feature_matrix\.zarr\.zip$</code> |  | Cell-feature matrix made by the Xenium onboard analysis pipeline. |
| <code>lab_processed\/xenium_bundle\/morphology_focus\/.*</code> |  | The morphology_focus/ subdirectory should include all morphology_focus.ome.tif files for Xenium datasets with multiple morphology focus files. |
| <code>lab_processed\/xenium_bundle\/morphology_focus\/morphology_focus_0000\.ome\.tif$</code> |  | Morphology focus file - DAPI image |
| <code>lab_processed\/xenium_bundle\/morphology_focus\/morphology_focus_0001\.ome\.tif$</code> |  | Morphology focus file - boundary (ATP1A1/E-Cadherin/CD45) image |
| <code>lab_processed\/xenium_bundle\/morphology_focus\/morphology_focus_0002\.ome\.tif$</code> |  | Morphology focus file - interior - RNA (18S) image |
| <code>lab_processed\/xenium_bundle\/morphology_focus\/morphology_focus_0003\.ome\.tif$</code> |  | Morphology focus file - interior - protein (alphaSMA/Vimentin) image |

<summary><b>Version 3.7</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/markers\.csv$</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection. This should minimally contain name of fluorophor, vendor, channel. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/custom_probe_set\.bed$</code> |  | This is a BED file version of the custom probe set file. |
| <code>raw\/transcript_locations\.csv$</code> | ‚úì | Contains decription of the location of all decoded transcripts. The origin of the coordinate is 0,0 at the top left corner of the image. The file should include: gene name, x, y, z (optional), quality score (optional). It is expected that the first row in the file contains the column header. |
| <code>raw\/custom_gene_list\.csv$</code> |  | This describes the target genes profiled by the assay. For advanced design, this can be probes sequences for splicing or other analysis for any target of interest. The format should minimally contain: gene name, ensemble ID |
| <code>raw\/probes\.csv$</code> |  | A CSV file describing the probe panel used. This is tyipcally what's used to specific the probe set when ordering a probe panel for a Xenium run. |
| <code>raw\/gene_panel\.json$</code> | ‚úì | This is the JSON file describing the probes, as output from the xenium-ranger pipeline. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tif&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*tissue-boundary\.geojson$</code> |  | **[QA/QC]** If the boundaries of the tissue have been identified (e.g., by manual efforts), then the boundary geometry can be included as a GeoJSON file named ‚Äú*.tissue-boundary.geojson‚Äù. |
| <code>lab_processed\/xenium_bundle\/.*</code> | ‚úì | The Xenium bundle includes *.zarr, *.parquet, *.tiff and *.gz files et al. Zarr files are required to view and explore data in Xenium Explorer. Parquet files are faster than CSV files and used extensively by 10X Genomics in Xenium and Visium HD. Sites are free to include in this subdirectory as many Xenium bundle files as possible: <https://www.10xgenomics.com/support/software/xenium-onboard-analysis/latest/analysis/xoa-output-understanding-outputs> |
| <code>lab_processed\/xenium_bundle\/cell_feature_matrix\.h5$</code> | ‚úì | Contains cell feature matrix |
| <code>lab_processed\/xenium_bundle\/experiment\.xenium$</code> | ‚úì | Contains specifications |
| <code>lab_processed\/xenium_bundle\/nucleus_boundaries\.parquet$</code> | ‚úì | Contains polygons of nucleus boundaries |
| <code>lab_processed\/xenium_bundle\/cell_boundaries\.parquet$</code> | ‚úì | Contains polygons of cell boundaries |
| <code>lab_processed\/xenium_bundle\/transcripts\.parquet$</code> | ‚úì | Contains transcripts |
| <code>lab_processed\/xenium_bundle\/cells\.parquet$</code> | ‚úì | Contains cell metadata |
| <code>lab_processed\/xenium_bundle\/morphology_mip\.ome\.tif$</code> |  | Contains morphology mip |
| <code>lab_processed\/xenium_bundle\/morphology_focus\.ome\.tif$</code> |  | Morphology focus file for Xenium datasets with a single morphology focus file |
| <code>lab_processed\/xenium_bundle\/morphology_focus\/.*</code> |  | The morphology_focus/ subdirectory should include all morphology_focus.ome.tif files for Xenium datasets with multiple morphology focus files. |
| <code>lab_processed\/xenium_bundle\/morphology_focus\/morphology_focus_0000\.ome\.tif$</code> |  | Morphology focus file - DAPI image |
| <code>lab_processed\/xenium_bundle\/morphology_focus\/morphology_focus_0001\.ome\.tif$</code> |  | Morphology focus file - boundary (ATP1A1/E-Cadherin/CD45) image |
| <code>lab_processed\/xenium_bundle\/morphology_focus\/morphology_focus_0002\.ome\.tif$</code> |  | Morphology focus file - interior - RNA (18S) image |
| <code>lab_processed\/xenium_bundle\/morphology_focus\/morphology_focus_0003\.ome\.tif$</code> |  | Morphology focus file - interior - protein (alphaSMA/Vimentin) image |

<summary><b>Version 3.6</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/markers\.csv$</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection. This should minimally contain name of fluorophor, vendor, channel. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/custom_probe_set\.bed$</code> |  | This is a BED file version of the custom probe set file. |
| <code>raw\/transcript_locations\.csv$</code> | ‚úì | Contains decription of the location of all decoded transcripts. The origin of the coordinate is 0,0 at the top left corner of the image. The file should include: gene name, x, y, z (optional), quality score (optional). It is expected that the first row in the file contains the column header. |
| <code>raw\/custom_gene_list\.csv$</code> |  | This describes the target genes profiled by the assay. For advanced design, this can be probes sequences for splicing or other analysis for any target of interest. The format should minimally contain: gene name, ensemble ID |
| <code>raw\/probes\.csv$</code> |  | A CSV file describing the probe panel used. This is tyipcally what's used to specific the probe set when ordering a probe panel for a Xenium run. |
| <code>raw\/gene_panel\.json$</code> | ‚úì | This is the JSON file describing the probes, as output from the xenium-ranger pipeline. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*tissue-boundary\.geojson$</code> |  | **[QA/QC]** If the boundaries of the tissue have been identified (e.g., by manual efforts), then the boundary geometry can be included as a GeoJSON file named ‚Äú*.tissue-boundary.geojson‚Äù. |
| <code>lab_processed\/xenium_bundle\/.*</code> | ‚úì | The Xenium bundle includes *.zarr, *.parquet, *.tiff and *.gz files et al. Zarr files are required to view and explore data in Xenium Explorer. Parquet files are faster than CSV files and used extensively by 10X Genomics in Xenium and Visium HD. Sites are free to include in this subdirectory as many Xenium bundle files as possible: <https://www.10xgenomics.com/support/software/xenium-onboard-analysis/latest/analysis/xoa-output-understanding-outputs> |
| <code>lab_processed\/xenium_bundle\/cell_feature_matrix\.h5$</code> | ‚úì | Contains cell feature matrix |
| <code>lab_processed\/xenium_bundle\/experiment\.xenium$</code> | ‚úì | Contains specifications |
| <code>lab_processed\/xenium_bundle\/nucleus_boundaries\.parquet$</code> | ‚úì | Contains polygons of nucleus boundaries |
| <code>lab_processed\/xenium_bundle\/cell_boundaries\.parquet$</code> | ‚úì | Contains polygons of cell boundaries |
| <code>lab_processed\/xenium_bundle\/transcripts\.parquet$</code> | ‚úì | Contains transcripts |
| <code>lab_processed\/xenium_bundle\/cells\.parquet$</code> | ‚úì | Contains cell metadata |
| <code>lab_processed\/xenium_bundle\/morphology_mip\.ome\.tif$</code> |  | Contains morphology mip |
| <code>lab_processed\/xenium_bundle\/morphology_focus[^\/]*\.ome\.tif$</code> | ‚úì | Contains morphology focus |

<summary><b>Version 3.5</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/markers\.csv$</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection. This should minimally contain name of fluorophor, vendor, channel. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/custom_probe_set\.bed$</code> |  | This is a BED file version of the custom probe set file. |
| <code>raw\/transcript_locations\.csv$</code> | ‚úì | Contains decription of the location of all decoded transcripts. The origin of the coordinate is 0,0 at the top left corner of the image. The file should include: gene name, x, y, z (optional), quality score (optional). It is expected that the first row in the file contains the column header. |
| <code>raw\/custom_gene_list\.csv$</code> |  | This describes the target genes profiled by the assay. For advanced design, this can be probes sequences for splicing or other analysis for any target of interest. The format should minimally contain: gene name, ensemble ID |
| <code>raw\/probes\.csv$</code> |  | A CSV file describing the probe panel used. This is tyipcally what's used to specific the probe set when ordering a probe panel for a Xenium run. |
| <code>raw\/gene_panel\.json$</code> | ‚úì | This is the JSON file describing the probes, as output from the xenium-ranger pipeline. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*tissue-boundary\.geojson$</code> |  | **[QA/QC]** If the boundaries of the tissue have been identified (e.g., by manual efforts), then the boundary geometry can be included as a GeoJSON file named ‚Äú*.tissue-boundary.geojson‚Äù. |
| <code>lab_processed\/xenium_bundle\/.*</code> | ‚úì | The Xenium bundle includes *.zarr, *.parquet, *.tiff and *.gz files et al. Zarr files are required to view and explore data in Xenium Explorer. Parquet files are faster than CSV files and used extensively by 10X Genomics in Xenium and Visium HD. Sites are free to include in this subdirectory as many Xenium bundle files as possible: <https://www.10xgenomics.com/support/software/xenium-onboard-analysis/latest/analysis/xoa-output-understanding-outputs> |
| <code>lab_processed\/xenium_bundle\/cell_feature_matrix\.h5$</code> | ‚úì | Contains cell feature matrix |
| <code>lab_processed\/xenium_bundle\/experiment\.xenium$</code> | ‚úì | Contains specifications |
| <code>lab_processed\/xenium_bundle\/nucleus_boundaries\.parquet$</code> | ‚úì | Contains polygons of nucleus boundaries |
| <code>lab_processed\/xenium_bundle\/cell_boundaries\.parquet$</code> | ‚úì | Contains polygons of cell boundaries |
| <code>lab_processed\/xenium_bundle\/transcripts\.parquet$</code> | ‚úì | Contains transcripts |
| <code>lab_processed\/xenium_bundle\/cells\.parquet$</code> | ‚úì | Contains cell metadata |
| <code>lab_processed\/xenium_bundle\/morphology_mip\.ome\.tif$</code> |  | Contains morphology mip |
| <code>lab_processed\/xenium_bundle\/morphology_focus\.ome\.tif$</code> | ‚úì | Contains morphology focus |

<summary><b>Version 3.4</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/markers\.csv$</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection. This should minimally contain name of fluorophor, vendor, channel. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/custom_probe_set\.bed$</code> |  | This is a BED file version of the custom probe set file. |
| <code>raw\/transcript_locations\.csv$</code> | ‚úì | Contains decription of the location of all decoded transcripts. The origin of the coordinate is 0,0 at the top left corner of the image. The file should include: gene name, x, y, z (optional), quality score (optional). It is expected that the first row in the file contains the column header. |
| <code>raw\/custom_gene_list\.csv$</code> |  | This describes the target genes profiled by the assay. For advanced design, this can be probes sequences for splicing or other analysis for any target of interest. The format should minimally contain: gene name, ensemble ID |
| <code>raw\/probes\.csv$</code> |  | A CSV file describing the probe panel used. This is tyipcally what's used to specific the probe set when ordering a probe panel for a Xenium run. |
| <code>raw\/gene_panel\.json$</code> | ‚úì | This is the JSON file describing the probes, as output from the xenium-ranger pipeline. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*tissue-boundary\.geojson$</code> |  | **[QA/QC]** If the boundaries of the tissue have been identified (e.g., by manual efforts), then the boundary geometry can be included as a GeoJSON file named ‚Äú*.tissue-boundary.geojson‚Äù. |
| <code>lab_processed\/xenium_bundle\/.*</code> | ‚úì | The Xenium bundle includes *.zarr, *.parquet, *.tiff and *.gz files et al. Zarr files are required to view and explore data in Xenium Explorer. Parquet files are faster than CSV files and used extensively by 10X Genomics in Xenium and Visium HD. Sites are free to include in this subdirectory as many Xenium bundle files as possible: <https://www.10xgenomics.com/support/software/xenium-onboard-analysis/latest/analysis/xoa-output-understanding-outputs> |
| <code>lab_processed\/xenium_bundle\/cell_feature_matrix\.h5$</code> | ‚úì | Contains cell feature matrix |
| <code>lab_processed\/xenium_bundle\/experiment\.xenium$</code> | ‚úì | Contains specifications |
| <code>lab_processed\/xenium_bundle\/nucleus_boundaries\.parquet$</code> | ‚úì | Contains polygons of nucleus boundaries |
| <code>lab_processed\/xenium_bundle\/cell_boundaries\.parquet$</code> | ‚úì | Contains polygons of cell boundaries |
| <code>lab_processed\/xenium_bundle\/transcripts\.parquet$</code> | ‚úì | Contains transcripts |
| <code>lab_processed\/xenium_bundle\/cells\.parquet$</code> | ‚úì | Contains cell metadata |
| <code>lab_processed\/xenium_bundle\/morphology_mip\.ome\.tif$</code> | ‚úì | Contains morphology mip |
| <code>lab_processed\/xenium_bundle\/morphology_focus\.ome\.tif$</code> | ‚úì | Contains morphology focus |

<summary><b>Version 3.3</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/markers\.csv$</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection. This should minimally contain name of fluorophor, vendor, channel. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/custom_probe_set\.bed$</code> |  | This is a BED file version of the custom probe set file. |
| <code>raw\/transcript_locations\.csv$</code> | ‚úì | Contains decription of the location of all decoded transcripts. The origin of the coordinate is 0,0 at the top left corner of the image. The file should include: gene name, x, y, z (optional), quality score (optional). It is expected that the first row in the file contains the column header. |
| <code>raw\/custom_gene_list\.csv$</code> |  | This describes the target genes profiled by the assay. For advanced design, this can be probes sequences for splicing or other analysis for any target of interest. The format should minimally contain: gene name, ensemble ID |
| <code>raw\/probes\.csv$</code> |  | A CSV file describing the probe panel used. This is tyipcally what's used to specific the probe set when ordering a probe panel for a Xenium run. |
| <code>raw\/gene_panel\.json$</code> | ‚úì | This is the JSON file describing the probes, as output from the xenium-ranger pipeline. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/xenium_bundle\/.*</code> | ‚úì | The Xenium bundle includes *.zarr, *.parquet, *.tiff and *.gz files et al. Zarr files are required to view and explore data in Xenium Explorer. Parquet files are faster than CSV files and used extensively by 10X Genomics in Xenium and Visium HD. Sites are free to include in this subdirectory as many Xenium bundle files as possible: <https://www.10xgenomics.com/support/software/xenium-onboard-analysis/latest/analysis/xoa-output-understanding-outputs> |
| <code>lab_processed\/xenium_bundle\/cell_feature_matrix\.h5$</code> | ‚úì | Contains cell feature matrix |
| <code>lab_processed\/xenium_bundle\/experiment\.xenium$</code> | ‚úì | Contains specifications |
| <code>lab_processed\/xenium_bundle\/nucleus_boundaries\.parquet$</code> | ‚úì | Contains polygons of nucleus boundaries |
| <code>lab_processed\/xenium_bundle\/cell_boundaries\.parquet$</code> | ‚úì | Contains polygons of cell boundaries |
| <code>lab_processed\/xenium_bundle\/transcripts\.parquet$</code> | ‚úì | Contains transcripts |
| <code>lab_processed\/xenium_bundle\/cells\.parquet$</code> | ‚úì | Contains cell metadata |
| <code>lab_processed\/xenium_bundle\/morphology_mip\.ome\.tif$</code> | ‚úì | Contains morphology mip |
| <code>lab_processed\/xenium_bundle\/morphology_focus\.ome\.tif$</code> | ‚úì | Contains morphology focus |

<summary><b>Version 3.2</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/markers\.csv$</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/custom_probe_set\.bed$</code> |  | This is a BED file version of the custom probe set file. |
| <code>raw\/transcript_locations\.csv$</code> | ‚úì | The origin of the coordinate is 0,0 at the top left corner of the image. The file should include: gene name, x, y, z (optional), quality score (optional). It is expected that the first row in the file contains the column header. |
| <code>raw\/custom_gene_list\.csv$</code> |  | This describes the target genes profiled by the assay. For advanced design, this can be probes sequences for splicing or other analysis for any target of interest. The format should minimally contain: gene name, ensemble ID |
| <code>raw\/probes\.csv$</code> |  | A CSV file describing the probe panel used. This is tyipcally what's used to specific the probe set when ordering a probe panel for a Xenium run. |
| <code>raw\/gene_panel\.json$</code> | ‚úì | This is the JSON file describing the probes, as output from the xenium-ranger pipeline. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/xenium_bundle\/.*</code> | ‚úì | The Xenium bundle includes *.zarr, *.parquet, *.tiff and *.gz files et al. Zarr files are required to view and explore data in Xenium Explorer. Parquet files are faster than CSV files and used extensively by 10X Genomics in Xenium and Visium HD. Sites are free to include in this subdirectory as many Xenium bundle files as possible: <https://www.10xgenomics.com/support/software/xenium-onboard-analysis/latest/analysis/xoa-output-understanding-outputs> |
| <code>lab_processed\/xenium_bundle\/cell_feature_matrix\.h5$</code> | ‚úì | Contains cell feature matrix |

<summary><b>Version 3.1</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/markers\.csv$</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/custom_probe_set\.bed$</code> |  | This is a BED file version of the custom probe set file. |
| <code>raw\/transcript_locations\.csv$</code> | ‚úì | The origin of the coordinate is 0,0 at the top left corner of the image. The file should include: gene name, x, y, z (optional), quality score (optional). It is expected that the first row in the file contains the column header. |
| <code>raw\/custom_gene_list\.csv$</code> |  | This describes the target genes profiled by the assay. For advanced design, this can be probes sequences for splicing or other analysis for any target of interest. The format should minimally contain: gene name, ensemble ID |
| <code>raw\/probes\.csv$</code> |  | A CSV file describing the probe panel used. This is tyipcally what's used to specific the probe set when ordering a probe panel for a Xenium run. |
| <code>raw\/gene_panel\.json$</code> | ‚úì | This is the JSON file describing the probes, as output from the xenium-ranger pipeline. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff&#124;tif)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.(?:tif&#124;tiff)$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-(?:tif&#124;tiff)\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/xenium_bundle\/.*</code> | ‚úì | The Xenium bundle includes *.zarr, *.parquet, *.tiff and *.gz files et al. Zarr files are required to view and explore data in Xenium Explorer. Parquet files are faster than CSV files and used extensively by 10X Genomics in Xenium and Visium HD. Sites are free to include in this subdirectory as many Xenium bundle files as possible: <https://www.10xgenomics.com/support/software/xenium-onboard-analysis/latest/analysis/xoa-output-understanding-outputs> |
| <code>lab_processed\/xenium_bundle\/cell_feature_matrix\.h5$</code> | ‚úì | Contains cell feature matrix |

<summary><b>Version 3.0</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/markers\.csv$</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/custom_probe_set\.bed$</code> |  | This is a BED file version of the custom probe set file. |
| <code>raw\/transcript_locations\.csv$</code> | ‚úì | The origin of the coordinate is 0,0 at the top left corner of the image. The file should include: gene name, x, y, z (optional), quality score (optional). It is expected that the first row in the file contains the column header. |
| <code>raw\/custom_gene_list\.csv$</code> |  | This describes the target genes profiled by the assay. For advanced design, this can be probes sequences for splicing or other analysis for any target of interest. The format should minimally contain: gene name, ensemble ID |
| <code>raw\/probes\.csv$</code> |  | A CSV file describing the probe panel used. This is tyipcally what's used to specific the probe set when ordering a probe panel for a Xenium run. |
| <code>raw\/gene_panel\.json$</code> | ‚úì | This is the JSON file describing the probes, as output from the xenium-ranger pipeline. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/xenium_bundle\/.*</code> | ‚úì | The Xenium bundle includes *.zarr, *.parquet, *.tiff and *.gz files et al. Zarr files are required to view and explore data in Xenium Explorer. Parquet files are faster than CSV files and used extensively by 10X Genomics in Xenium and Visium HD. Sites should include all Xenium bundle files in this subdirectory. Xenium bundle files: <https://www.10xgenomics.com/support/software/xenium-onboard-analysis/latest/analysis/xoa-output-understanding-outputs> |
| <code>lab_processed\/xenium_bundle\/cell_feature_matrix\.h5$</code> | ‚úì | Contains cell feature matrix |
| <code>lab_processed\/xenium_bundle\/experiment\.xenium$</code> | ‚úì | Contains specifications. |
| <code>lab_processed\/xenium_bundle\/nucleus_boundaries\.parquet$</code> | ‚úì | Contains polygons of nucleus boundaries. |
| <code>lab_processed\/xenium_bundle\/cell_boundaries\.parquet$</code> | ‚úì | Contains polygons of cell boundaries. |
| <code>lab_processed\/xenium_bundle\/transcripts\.parquet$</code> | ‚úì | Contains transcripts |
| <code>lab_processed\/xenium_bundle\/cells\.parquet$</code> | ‚úì | Contains cell metadata |
| <code>lab_processed\/xenium_bundle\/morphology_mip\.ome\.tif$</code> | ‚úì | Contains morphology mip |
| <code>lab_processed\/xenium_bundle\/morphology_focus\.ome\.tif$</code> | ‚úì | Contains morphology focus |

<summary><b>Version 2.4</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/markers\.csv$</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/custom_probe_set\.bed$</code> |  | This is a BED file version of the custom probe set file. |
| <code>raw\/transcript_locations\.csv$</code> | ‚úì | The origin of the coordinate is 0,0 at the top left corner of the image. The file should include: gene name, x, y, z (optional), quality score (optional). It is expected that the first row in the file contains the column header. |
| <code>raw\/custom_gene_list\.csv$</code> |  | This describes the target genes profiled by the assay. For advanced design, this can be probes sequences for splicing or other analysis for any target of interest. The format should minimally contain: gene name, ensemble ID |
| <code>raw\/probes\.csv$</code> |  | A CSV file describing the probe panel used. This is tyipcally what's used to specific the probe set when ordering a probe panel for a Xenium run. |
| <code>raw\/gene_panel\.json$</code> | ‚úì | This is the JSON file describing the probes, as output from the xenium-ranger pipeline. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/xenium_bundle\/.*</code> | ‚úì | The Xenium bundle includes *.zarr, *.parquet, *.tiff and *.gz files et al. Zarr files are required to view and explore data in Xenium Explorer. Parquet files are faster than CSV files and used extensively by 10X Genomics in Xenium and Visium HD. Sites are free to include in this subdirectory as many Xenium bundle files as possible: <https://www.10xgenomics.com/support/software/xenium-onboard-analysis/latest/analysis/xoa-output-understanding-outputs> |
| <code>lab_processed\/xenium_bundle\/cell_feature_matrix\.h5$</code> | ‚úì | Contains cell feature matrix |

<summary><b>Version 2.3</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/markers\.csv$</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/custom_probe_set\.bed$</code> |  | This is a BED file version of the custom probe set file. |
| <code>raw\/transcript_locations\.csv$</code> | ‚úì | The origin of the coordinate is 0,0 at the top left corner of the image. The file should include: gene name, x, y, z (optional), quality score (optional). It is expected that the first row in the file contains the column header. |
| <code>raw\/custom_gene_list\.csv$</code> |  | This describes the target genes profiled by the assay. For advanced design, this can be probes sequences for splicing or other analysis for any target of interest. The format should minimally contain: gene name, ensemble ID |
| <code>raw\/probes\.csv$</code> |  | A CSV file describing the probe panel used. This is tyipcally what's used to specific the probe set when ordering a probe panel for a Xenium run. |
| <code>raw\/gene_panel\.json$</code> | ‚úì | This is the JSON file describing the probes, as output from the xenium-ranger pipeline. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/xenium_bundle\/.*</code> | ‚úì | The Xenium bundle includes *.zarr, *.parquet, *.tiff and *.gz files et al. Zarr files are required to view and explore data in Xenium Explorer. Parquet files are faster than CSV files and used extensively by 10X Genomics in Xenium and Visium HD. Sites are free to include in this subdirectory as many Xenium bundle files as possible: <https://www.10xgenomics.com/support/software/xenium-onboard-analysis/latest/analysis/xoa-output-understanding-outputs> |

<summary><b>Version 2.2</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/markers\.csv$</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/custom_probe_set\.bed$</code> |  | This is a BED file version of the custom probe set file. |
| <code>raw\/transcript_locations\.csv$</code> | ‚úì | The origin of the coordinate is 0,0 at the top left corner of the image. The file should include: gene name, x, y, z (optional), quality score (optional). It is expected that the first row in the file contains the column header. |
| <code>raw\/custom_gene_list\.csv$</code> |  | This describes the target genes profiled by the assay. For advanced design, this can be probes sequences for splicing or other analysis for any target of interest. The format should minimally contain: gene name, ensemble ID |
| <code>raw\/probes\.csv$</code> |  | A CSV file describing the probe panel used. This is tyipcally what's used to specific the probe set when ordering a probe panel for a Xenium run. |
| <code>raw\/gene_panel\.json$</code> | ‚úì | This is the JSON file describing the probes, as output from the xenium-ranger pipeline. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/xenium_bundle\/.*</code> |  | The Xenium bundle includes *.zarr, *.parquet, *.tiff and *.gz files et al. Zarr files are required to view and explore data in Xenium Explorer. Parquet files are faster than CSV files and used extensively by 10X Genomics in Xenium and Visium HD. Sites are free to include in this subdirectory as many Xenium bundle files as possible: <https://www.10xgenomics.com/support/software/xenium-onboard-analysis/latest/analysis/xoa-output-understanding-outputs> |

<summary><b>Version 2.1</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/markers\.csv</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection. |
| <code>raw\/additional_panels_used\.csv</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/custom_probe_set\.bed</code> |  | This is a BED file version of the custom probe set file. |
| <code>raw\/transcript_locations\.csv</code> | ‚úì | The origin of the coordinate is 0,0 at the top left corner of the image. The file should include: gene name, x, y, z (optional), quality score (optional). It is expected that the first row in the file contains the column header. |
| <code>raw\/custom_gene_list\.csv</code> |  | This describes the target genes profiled by the assay. For advanced design, this can be probes sequences for splicing or other analysis for any target of interest. The format should minimally contain: gene name, ensemble ID |
| <code>raw\/probes\.csv</code> |  | A CSV file describing the probe panel used. This is tyipcally what's used to specific the probe set when ordering a probe panel for a Xenium run. |
| <code>raw\/gene_panel\.json</code> | ‚úì | This is the JSON file describing the probes, as output from the xenium-ranger pipeline. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/xenium_bundle\/.*</code> |  | The Xenium bundle includes *.zarr, *.parquet, *.tiff and *.gz files et al. Zarr files are required to view and explore data in Xenium Explorer. Parquet files are faster than CSV files and used extensively by 10X Genomics in Xenium and Visium HD. Sites are free to include in this subdirectory as many Xenium bundle files as possible: <https://www.10xgenomics.com/support/software/xenium-onboard-analysis/latest/analysis/xoa-output-understanding-outputs> |

<summary><b>Version 2.0</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/markers\.csv</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection. |
| <code>raw\/additional_panels_used\.csv</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/custom_probe_set\.bed</code> |  | This is a BED file version of the custom probe set file. |
| <code>raw\/transcript_locations\.csv</code> | ‚úì | The origin of the coordinate is 0,0 at the top left corner of the image. The file should include: gene name, x, y, z (optional), quality score (optional). It is expected that the first row in the file contains the column header. |
| <code>raw\/custom_gene_list\.csv</code> |  | This describes the target genes profiled by the assay. For advanced design, this can be probes sequences for splicing or other analysis for any target of interest. The format should minimally contain: gene name, ensemble ID |
| <code>raw\/probes\.csv</code> |  | A CSV file describing the probe panel used. This is tyipcally what's used to specific the probe set when ordering a probe panel for a Xenium run. |
| <code>raw\/gene_panel\.json</code> | ‚úì | This is the JSON file describing the probes, as output from the xenium-ranger pipeline. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |

