---
title: GeoMx (NGS)
schema_name: geomx-ngs
category: Spatial Transcriptomics
all_versions_deprecated: False
exclude_from_index: False
layout: default

---
Prepare your metadata based on the latest metadata schema using one of the template files below. See the instructions in the [Metadata Validation Workflow](https://docs.google.com/document/d/1lfgiDGbyO4K4Hz1FMsJjmJd9RdwjShtJqFYNwKpbcZY) document for more information on preparing and validating your metadata.tsv file prior to submission.

Related files:


- [üìù Excel template](https://raw.githubusercontent.com/hubmapconsortium/dataset-metadata-spreadsheet/main/geomx-ngs/latest/geomx-ngs.xlsx): For metadata entry.
- [üìù TSV template](https://raw.githubusercontent.com/hubmapconsortium/dataset-metadata-spreadsheet/main/geomx-ngs/latest/geomx-ngs.tsv): Alternative for metadata entry.


REQUIRED - For this assay, you must also prepare and submit an additional metadata.tsv file following the metadata schema linked here for [RNAseq (with probes)](https://hubmapconsortium.github.io/ingest-validation-tools/rnaseq-with-probes/current/).

## Metadata schema


<summary><a href="https://openview.metadatacenter.org/templates/https:%2F%2Frepo.metadatacenter.org%2Ftemplates%2F11687c2e-1fbe-4405-a73c-e7ce3294042f"><b>Version 2 (use this one)</b></a></summary>



<br>

## Directory schemas
<summary><b>Version 2.10 (use this one)</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/[^\/]+_LabWorksheet.txt$</code> | ‚úì | An Excel spreadsheet to refer to in setting up the library. This file documents all of the samples from a single collection plate. Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_config\.ini$</code> | ‚úì | Needed to generate the DCC file from the fastq file. Contains pipeline processing parameters.  Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_SeqCodeIndices\.csv$</code> |  | A file with sample information needed by the Illumina software. Use the contents of the SeqCodeIndices.csv file to create a SampleSheet.csv for input to the Illumina sequencer. (NextSeq 1000/2000 users download a SampleSheet.csv and whitelist.txt instead of SeqCodeIndices.csv.)  Generated by DSP run. |
| <code>raw\/[^\/]+_SampleSheet\.csv$</code> |  | Used by NextSeq 1000/2000 users, along with whitelist.txt, in place of SeqCodeIndices.csv |
| <code>raw\/[^\/]+_whitelist\.txt$</code> |  | Used by NextSeq 1000/2000 users, along with SampleSheet.csv, in place of SeqCodeIndices.csv |
| <code>raw\/markers\.csv$</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection [this should be similar in structure to the antibodies file] |
| <code>raw\/[^\/]*targets\.pkc$</code> | ‚úì | The file listing probe barcode sequence and corresponding gene symbol or proteins targeted by that probe. This should be consistent for the same probe panel. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include: manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include: target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/fastq\/.*</code> | ‚úì | Raw sequencing files for the experiment |
| <code>raw\/fastq\/oligo\/.*</code> | ‚úì | Directory containing fastq files pertaining to oligo sequencing. |
| <code>raw\/fastq\/oligo\/[^\/]+\.fastq\.gz$</code> | ‚úì | This is a gzip version of the fastq file. This file contains the cell barcode and unique molecular identifier (technical). |
| <code>raw\/images\/.*</code> |  | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/Initial\s{1}Dataset\.xlsx$</code> | ‚úì | **[QA/QC]** An excel spreadsheet that is downloaded from the GeoMx DSP Data Analysis Suite containing QA/QC metrics based on raw, unprocessed target counts. This file contains one row per AOI/segment and no analyses span AOI. The AOIs included in this file can come from different GeoMx runs and hence span Globus uploads. So care must be taken to make sure the appropriate AOIs are included in the file. |
| <code>lab_processed\/annotations\.xlsx$</code> |  | AOI specific annotations. This might include cell type and anatomical information. |
| <code>lab_processed\/dcc\/.*</code> | ‚úì | DCC files generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/dcc\/[^\/]+\.dcc$</code> | ‚úì | DCC files containing target probe counts, generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*\.tissue-boundary\.geojson$</code> |  | **[QA/QC]** If the boundaries of the tissue have been identified (e.g., by manual efforts), then the boundary geometry can be included as a GeoJSON file named ‚Äú*.tissue-boundary.geojson‚Äù. |
| <code>lab_processed\/primary_analysis\/.*</code> | ‚úì | Primary analysis results |
| <code>lab_processed\/primary_analysis\/[^\/]+\.xlsx$</code> |  | **[QA/QC]** File containing results from initial processing by GeoMx DSP Data Analysis Suite including optional removal of segments and targets based on QC flags and LOQ and normalization using Q3 normalization. This file will not be available for sites that do not use the GeoMx DSP Data Analysis Suite. |

<summary><b>Version 2.9</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/[^\/]+_LabWorksheet.txt$</code> | ‚úì | An Excel spreadsheet to refer to in setting up the library. This file documents all of the samples from a single collection plate. Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_config\.ini$</code> | ‚úì | Needed to generate the DCC file from the fastq file. Contains pipeline processing parameters.  Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_SeqCodeIndices\.csv$</code> |  | A file with sample information needed by the Illumina software. Use the contents of the SeqCodeIndices.csv file to create a SampleSheet.csv for input to the Illumina sequencer. (NextSeq 1000/2000 users download a SampleSheet.csv and whitelist.txt instead of SeqCodeIndices.csv.)  Generated by DSP run. |
| <code>raw\/[^\/]+_SampleSheet\.csv$</code> |  | Used by NextSeq 1000/2000 users, along with whitelist.txt, in place of SeqCodeIndices.csv |
| <code>raw\/[^\/]+_whitelist\.txt$</code> |  | Used by NextSeq 1000/2000 users, along with SampleSheet.csv, in place of SeqCodeIndices.csv |
| <code>raw\/markers\.csv$</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection [this should be similar in structure to the antibodies file] |
| <code>raw\/[^\/]*targets\.pkc$</code> | ‚úì | The file listing probe barcode sequence and corresponding gene symbol or proteins targeted by that probe. This should be consistent for the same probe panel. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include: manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include: target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/fastq\/.*</code> | ‚úì | Raw sequencing files for the experiment |
| <code>raw\/fastq\/oligo\/.*</code> | ‚úì | Directory containing fastq files pertaining to oligo sequencing. |
| <code>raw\/fastq\/oligo\/[^\/]+\.fastq\.gz$</code> | ‚úì | This is a gzip version of the fastq file. This file contains the cell barcode and unique molecular identifier (technical). |
| <code>raw\/images\/.*</code> |  | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/Initial\s{1}Dataset\.xlsx$</code> | ‚úì | **[QA/QC]** An excel spreadsheet that is downloaded from the GeoMx DSP Data Analysis Suite containing QA/QC metrics based on raw, unprocessed target counts. This file contains one row per AOI/segment and no analyses span AOI. The AOIs included in this file can come from different GeoMx runs and hence span Globus uploads. So care must be taken to make sure the appropriate AOIs are included in the file. |
| <code>lab_processed\/annotations\.xlsx$</code> |  | AOI specific annotations. This might include cell type and anatomical information. |
| <code>lab_processed\/dcc\/.*</code> | ‚úì | DCC files generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/dcc\/[^\/]+\.dcc$</code> | ‚úì | DCC files containing target probe counts, generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/primary_analysis\/.*</code> | ‚úì | Primary analysis results |
| <code>lab_processed\/primary_analysis\/[^\/]+\.xlsx$</code> |  | **[QA/QC]** File containing results from initial processing by GeoMx DSP Data Analysis Suite including optional removal of segments and targets based on QC flags and LOQ and normalization using Q3 normalization. This file will not be available for sites that do not use the GeoMx DSP Data Analysis Suite. |

<summary><b>Version 2.8</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/[^\/]+_LabWorksheet.txt$</code> | ‚úì | An Excel spreadsheet to refer to in setting up the library. This file documents all of the samples from a single collection plate. Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_config\.ini$</code> | ‚úì | Needed to generate the DCC file from the fastq file. Contains pipeline processing parameters.  Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_SeqCodeIndices\.csv$</code> |  | A file with sample information needed by the Illumina software. Use the contents of the SeqCodeIndices.csv file to create a SampleSheet.csv for input to the Illumina sequencer. (NextSeq 1000/2000 users download a SampleSheet.csv and whitelist.txt instead of SeqCodeIndices.csv.)  Generated by DSP run. |
| <code>raw\/[^\/]+_SampleSheet\.csv$</code> |  | Used by NextSeq 1000/2000 users, along with whitelist.txt, in place of SeqCodeIndices.csv |
| <code>raw\/[^\/]+_whitelist\.txt$</code> |  | Used by NextSeq 1000/2000 users, along with SampleSheet.csv, in place of SeqCodeIndices.csv |
| <code>raw\/markers\.csv$</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection [this should be similar in structure to the antibodies file] |
| <code>raw\/[^\/]*targets\.pkc$</code> | ‚úì | The file listing probe barcode sequence and corresponding gene symbol or proteins targeted by that probe. This should be consistent for the same probe panel. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include: manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include: target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/fastq\/.*</code> | ‚úì | Raw sequencing files for the experiment |
| <code>raw\/fastq\/oligo\/.*</code> | ‚úì | Directory containing fastq files pertaining to oligo sequencing. |
| <code>raw\/fastq\/oligo\/[^\/]+\.fastq\.gz$</code> | ‚úì | This is a gzip version of the fastq file. This file contains the cell barcode and unique molecular identifier (technical). |
| <code>raw\/images\/.*</code> |  | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/Initial\s{1}Dataset\.xlsx$</code> |  | **[QA/QC]** An excel spreadsheet that is downloaded from the GeoMx DSP Data Analysis Suite containing QA/QC metrics based on raw, unprocessed target counts. This file contains one row per AOI/segment and no analyses span AOI. The AOIs included in this file can come from different GeoMx runs and hence span Globus uploads. So care must be taken to make sure the appropriate AOIs are included in the file. This file will not be available for sites that do not use the GeoMx DSP Data Analysis Suite. |
| <code>lab_processed\/annotations\.xlsx$</code> |  | AOI specific annotations. This might include cell type and anatomical information. |
| <code>lab_processed\/dcc\/.*</code> | ‚úì | DCC files generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/dcc\/[^\/]+\.dcc$</code> | ‚úì | DCC files containing target probe counts, generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/primary_analysis\/.*</code> | ‚úì | Primary analysis results |
| <code>lab_processed\/primary_analysis\/[^\/]+\.xlsx$</code> | ‚úì | **[QA/QC]** File containing results from initial processing by GeoMx DSP Data Analysis Suite including optional removal of segments and targets based on QC flags and LOQ and normalization using Q3 normalization. |

<summary><b>Version 2.7</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/[^\/]+_LabWorksheet.txt$</code> | ‚úì | An Excel spreadsheet to refer to in setting up the library. This file documents all of the samples from a single collection plate. Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_config\.ini$</code> | ‚úì | Needed to generate the DCC file from the fastq file. Contains pipeline processing parameters.  Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_SeqCodeIndices\.csv$</code> |  | A file with sample information needed by the Illumina software. Use the contents of the SeqCodeIndices.csv file to create a SampleSheet.csv for input to the Illumina sequencer. (NextSeq 1000/2000 users download a SampleSheet.csv and whitelist.txt instead of SeqCodeIndices.csv.)  Generated by DSP run. |
| <code>raw\/[^\/]+_SampleSheet\.csv$</code> |  | Used by NextSeq 1000/2000 users, along with whitelist.txt, in place of SeqCodeIndices.csv |
| <code>raw\/[^\/]+_whitelist\.txt$</code> |  | Used by NextSeq 1000/2000 users, along with SampleSheet.csv, in place of SeqCodeIndices.csv |
| <code>raw\/markers\.csv$</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection [this should be similar in structure to the antibodies file] |
| <code>raw\/[^\/]*targets\.pkc$</code> | ‚úì | The file listing probe barcode sequence and corresponding gene symbol or proteins targeted by that probe. This should be consistent for the same probe panel. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include: manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include: target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/fastq\/.*</code> | ‚úì | Raw sequencing files for the experiment |
| <code>raw\/fastq\/oligo\/.*</code> | ‚úì | Directory containing fastq files pertaining to oligo sequencing. |
| <code>raw\/fastq\/oligo\/[^\/]+\.fastq\.gz$</code> | ‚úì | This is a gzip version of the fastq file. This file contains the cell barcode and unique molecular identifier (technical). |
| <code>raw\/images\/.*</code> |  | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/Initial\s{1}Dataset\.xlsx$</code> | ‚úì | **[QA/QC]** An excel spreadsheet that is downloaded from the GeoMx DSP Data Analysis Suite containing QA/QC metrics based on raw, unprocessed target counts. This file contains one row per AOI/segment and no analyses span AOI. The AOIs included in this file can come from different GeoMx runs and hence span Globus uploads. So care must be taken to make sure the appropriate AOIs are included in the file. |
| <code>lab_processed\/annotations\.xlsx$</code> |  | AOI specific annotations. This might include cell type and anatomical information. |
| <code>lab_processed\/dcc\/.*</code> | ‚úì | DCC files generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/dcc\/[^\/]+\.dcc$</code> | ‚úì | DCC files containing target probe counts, generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/primary_analysis\/.*</code> | ‚úì | Primary analysis results |
| <code>lab_processed\/primary_analysis\/[^\/]+\.xlsx$</code> | ‚úì | **[QA/QC]** File containing results from initial processing by GeoMx DSP Data Analysis Suite including optional removal of segments and targets based on QC flags and LOQ and normalization using Q3 normalization. |

<summary><b>Version 2.6</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/[^\/]+_LabWorksheet.txt$</code> | ‚úì | An Excel spreadsheet to refer to in setting up the library. This file documents all of the samples from a single collection plate. Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_config\.ini$</code> | ‚úì | Needed to generate the DCC file from the fastq file. Contains pipeline processing parameters.  Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_SeqCodeIndices\.csv$</code> |  | A file with sample information needed by the Illumina software. Use the contents of the SeqCodeIndices.csv file to create a SampleSheet.csv for input to the Illumina sequencer. (NextSeq 1000/2000 users download a SampleSheet.csv and whitelist.txt instead of SeqCodeIndices.csv.)  Generated by DSP run. |
| <code>raw\/[^\/]+_SampleSheet\.csv$</code> |  | Used by NextSeq 1000/2000 users, along with whitelist.txt, in place of SeqCodeIndices.csv |
| <code>raw\/[^\/]+_whitelist\.txt$</code> |  | Used by NextSeq 1000/2000 users, along with SampleSheet.csv, in place of SeqCodeIndices.csv |
| <code>raw\/markers\.csv$</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection [this should be similar in structure to the antibodies file] |
| <code>raw\/[^\/]*targets\.pkc$</code> | ‚úì | The file listing probe barcode sequence and corresponding gene symbol or proteins targeted by that probe. This should be consistent for the same probe panel. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include: manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include: target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/fastq\/.*</code> | ‚úì | Raw sequencing files for the experiment |
| <code>raw\/fastq\/oligo\/.*</code> | ‚úì | Directory containing fastq files pertaining to oligo sequencing. |
| <code>raw\/fastq\/oligo\/[^\/]+\.fastq\.gz$</code> | ‚úì | This is a gzip version of the fastq file. This file contains the cell barcode and unique molecular identifier (technical). |
| <code>raw\/images\/.*</code> |  | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff&#124;tif)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/Initial\s{1}Dataset\.xlsx$</code> | ‚úì | **[QA/QC]** An excel spreadsheet that is downloaded from the GeoMx DSP Data Analysis Suite containing QA/QC metrics based on raw, unprocessed target counts. This file contains one row per AOI/segment and no analyses span AOI. The AOIs included in this file can come from different GeoMx runs and hence span Globus uploads. So care must be taken to make sure the appropriate AOIs are included in the file. |
| <code>lab_processed\/annotations\.xlsx$</code> |  | AOI specific annotations. This might include cell type and anatomical information. |
| <code>lab_processed\/dcc\/.*</code> | ‚úì | DCC files generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/dcc\/[^\/]+\.dcc$</code> | ‚úì | DCC files containing target probe counts, generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.(?:tif&#124;tiff)$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-(?:tif&#124;tiff)\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/primary_analysis\/.*</code> | ‚úì | Primary analysis results |
| <code>lab_processed\/primary_analysis\/[^\/]+\.xlsx$</code> | ‚úì | **[QA/QC]** File containing results from initial processing by GeoMx DSP Data Analysis Suite including optional removal of segments and targets based on QC flags and LOQ and normalization using Q3 normalization. |

<summary><b>Version 2.5</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/[^\/]+_LabWorksheet.txt$</code> | ‚úì | An Excel spreadsheet to refer to in setting up the library. This file documents all of the samples from a single collection plate. Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_config\.ini$</code> | ‚úì | Needed to generate the DCC file from the fastq file. Contains pipeline processing parameters.  Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_SeqCodeIndices\.csv$</code> |  | A file with sample information needed by the Illumina software. Use the contents of the SeqCodeIndices.csv file to create a SampleSheet.csv for input to the Illumina sequencer. (NextSeq 1000/2000 users download a SampleSheet.csv and whitelist.txt instead of SeqCodeIndices.csv.)  Generated by DSP run. |
| <code>raw\/[^\/]+_SampleSheet\.csv$</code> |  | Used by NextSeq 1000/2000 users, along with whitelist.txt, in place of SeqCodeIndices.csv |
| <code>raw\/[^\/]+_whitelist\.txt$</code> |  | Used by NextSeq 1000/2000 users, along with SampleSheet.csv, in place of SeqCodeIndices.csv |
| <code>raw\/markers\.csv$</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection [this should be similar in structure to the antibodies file] |
| <code>raw\/[^\/]*targets\.pkc$</code> | ‚úì | The file listing probe barcode sequence and corresponding gene symbol or proteins targeted by that probe. This should be consistent for the same probe panel. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include: manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include: target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/fastq\/.*</code> | ‚úì | Raw sequencing files for the experiment |
| <code>raw\/fastq\/oligo\/.*</code> | ‚úì | Directory containing fastq files pertaining to oligo sequencing. |
| <code>raw\/fastq\/oligo\/[^\/]+\.fastq\.gz$</code> | ‚úì | This is a gzip version of the fastq file. This file contains the cell barcode and unique molecular identifier (technical). |
| <code>raw\/images\/.*</code> |  | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/Initial\s{1}Dataset\.xlsx$</code> | ‚úì | **[QA/QC]** An excel spreadsheet that is downloaded from the GeoMx DSP Data Analysis Suite containing QA/QC metrics based on raw, unprocessed target counts. This file contains one row per AOI/segment and no analyses span AOI. The AOIs included in this file can come from different GeoMx runs and hence span Globus uploads. So care must be taken to make sure the appropriate AOIs are included in the file. |
| <code>lab_processed\/annotations\.xlsx$</code> |  | AOI specific annotations. This might include cell type and anatomical information. |
| <code>lab_processed\/dcc\/.*</code> | ‚úì | DCC files generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/dcc\/[^\/]+\.dcc$</code> | ‚úì | DCC files containing target probe counts, generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/primary_analysis\/.*</code> | ‚úì | Primary analysis results |
| <code>lab_processed\/primary_analysis\/[^\/]+\.xlsx$</code> | ‚úì | **[QA/QC]** File containing results from initial processing by GeoMx DSP Data Analysis Suite including optional removal of segments and targets based on QC flags and LOQ and normalization using Q3 normalization. |

<summary><b>Version 2.4</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/[^\/]+_LabWorksheet.txt</code> | ‚úì | An Excel spreadsheet to refer to in setting up the library. This file documents all of the samples from a single collection plate. Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_config\.ini</code> | ‚úì | Needed to generate the DCC file from the fastq file. Contains pipeline processing parameters.  Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_SeqCodeIndices\.csv</code> |  | A file with sample information needed by the Illumina software. Use the contents of the SeqCodeIndices.csv file to create a SampleSheet.csv for input to the Illumina sequencer. (NextSeq 1000/2000 users download a SampleSheet.csv and whitelist.txt instead of SeqCodeIndices.csv.)  Generated by DSP run. |
| <code>raw\/[^\/]+_SampleSheet\.csv</code> |  | Used by NextSeq 1000/2000 users, along with whitelist.txt, in place of SeqCodeIndices.csv |
| <code>raw\/[^\/]+_whitelist\.txt</code> |  | Used by NextSeq 1000/2000 users, along with SampleSheet.csv, in place of SeqCodeIndices.csv |
| <code>raw\/markers\.csv</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection [this should be similar in structure to the antibodies file] |
| <code>raw\/[^\/]*targets\.pkc</code> | ‚úì | The file listing probe barcode sequence and corresponding gene symbol or proteins targeted by that probe. This should be consistent for the same probe panel. |
| <code>raw\/additional_panels_used\.csv</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/fastq\/.*</code> | ‚úì | Raw sequencing files for the experiment |
| <code>raw\/fastq\/oligo\/.*</code> | ‚úì | Directory containing fastq files pertaining to oligo sequencing. |
| <code>raw\/fastq\/oligo\/[^\/]+\.fastq\.gz</code> | ‚úì | This is a gzip version of the fastq file. This file contains the cell barcode and unique molecular identifier (technical). |
| <code>raw\/images\/.*</code> |  | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/Initial\s{1}Dataset\.xlsx</code> | ‚úì | **[QA/QC]** An excel spreadsheet that is downloaded from the GeoMx DSP Data Analysis Suite containing QA/QC metrics based on raw, unprocessed target counts. This file contains one row per AOI/segment and no analyses span AOI. The AOIs included in this file can come from different GeoMx runs and hence span Globus uploads. So care must be taken to make sure the appropriate AOIs are included in the file. |
| <code>lab_processed\/annotations\.xlsx</code> |  | AOI specific annotations. This might include cell type and anatomical information. |
| <code>lab_processed\/dcc\/.*</code> | ‚úì | DCC files generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/dcc\/[^\/]+\.dcc</code> | ‚úì | DCC files containing target probe counts, generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/primary_analysis\/.*</code> | ‚úì | Primary analysis results |
| <code>lab_processed\/primary_analysis\/[^\/]+\.xlsx</code> | ‚úì | **[QA/QC]** File containing results from initial processing by GeoMx DSP Data Analysis Suite including optional removal of segments and targets based on QC flags and LOQ and normalization using Q3 normalization. |

<summary><b>Version 2.3</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/[^\/]+_LabWorksheet.txt</code> | ‚úì | An Excel spreadsheet to refer to in setting up the library. This file documents all of the samples from a single collection plate. Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_config\.ini</code> | ‚úì | Needed to generate the DCC file from the fastq file. Contains pipeline processing parameters.  Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_SeqCodeIndices\.csv</code> | ‚úì | A file with sample information needed by the Illumina software. Use the contents of the SeqCodeIndices.csv file to create a SampleSheet.csv for input to the Illumina sequencer. (NextSeq 1000/2000 users download a SampleSheet.csv and whitelist.txt instead of SeqCodeIndices.csv.)  Generated by DSP run. |
| <code>raw\/markers\.csv</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection [this should be similar in structure to the antibodies file] |
| <code>raw\/[^\/]*targets\.pkc</code> | ‚úì | The file listing probe barcode sequence and corresponding gene symbol or proteins targeted by that probe. This should be consistent for the same probe panel. |
| <code>raw\/additional_panels_used\.csv</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/fastq\/.*</code> | ‚úì | Raw sequencing files for the experiment |
| <code>raw\/fastq\/oligo\/.*</code> | ‚úì | Directory containing fastq files pertaining to oligo sequencing. |
| <code>raw\/fastq\/oligo\/[^\/]+\.fastq\.gz</code> | ‚úì | This is a gzip version of the fastq file. This file contains the cell barcode and unique molecular identifier (technical). |
| <code>raw\/images\/.*</code> |  | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/Initial\s{1}Dataset\.xlsx</code> | ‚úì | **[QA/QC]** An excel spreadsheet that is downloaded from the GeoMx DSP Data Analysis Suite containing QA/QC metrics based on raw, unprocessed target counts. This file contains one row per AOI/segment and no analyses span AOI. The AOIs included in this file can come from different GeoMx runs and hence span Globus uploads. So care must be taken to make sure the appropriate AOIs are included in the file. |
| <code>lab_processed\/annotations\.xlsx</code> |  | AOI specific annotations. This might include cell type and anatomical information. |
| <code>lab_processed\/dcc\/.*</code> | ‚úì | DCC files generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/dcc\/[^\/]+\.dcc</code> | ‚úì | DCC files containing target probe counts, generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/primary_analysis\/.*</code> | ‚úì | Primary analysis results |
| <code>lab_processed\/primary_analysis\/[^\/]+\.xlsx</code> | ‚úì | **[QA/QC]** File containing results from initial processing by GeoMx DSP Data Analysis Suite including optional removal of segments and targets based on QC flags and LOQ and normalization using Q3 normalization. |

<summary><b>Version 2.2</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/[^\/]+_LabWorksheet.txt</code> | ‚úì | An Excel spreadsheet to refer to in setting up the library. This file documents all of the samples from a single collection plate. Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_config\.ini</code> | ‚úì | Needed to generate the DCC file from the fastq file. Contains pipeline processing parameters.  Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_SeqCodeIndices\.csv</code> | ‚úì | A file with sample information needed by the Illumina software. Use the contents of the SeqCodeIndices.csv file to create a SampleSheet.csv for input to the Illumina sequencer. (NextSeq 1000/2000 users download a SampleSheet.csv and whitelist.txt instead of SeqCodeIndices.csv.)  Generated by DSP run. |
| <code>raw\/markers\.csv</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection [this should be similar in structure to the antibodies file] |
| <code>raw\/[^\/]*targets\.pkc</code> | ‚úì | The file listing probe barcode sequence and corresponding gene symbol or proteins targeted by that probe. This should be consistent for the same probe panel. |
| <code>raw\/additional_panels_used\.csv</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv</code> | ‚úì | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/fastq\/.*</code> | ‚úì | Raw sequencing files for the experiment |
| <code>raw\/fastq\/oligo\/.*</code> | ‚úì | Directory containing fastq files pertaining to oligo sequencing. |
| <code>raw\/fastq\/oligo\/[^\/]+\.fastq\.gz</code> | ‚úì | This is a gzip version of the fastq file. This file contains the cell barcode and unique molecular identifier (technical). |
| <code>raw\/images\/.*</code> |  | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/Initial\s{1}Dataset\.xlsx</code> | ‚úì | **[QA/QC]** An excel spreadsheet that is downloaded from the GeoMx DSP Data Analysis Suite containing QA/QC metrics based on raw, unprocessed target counts. This file contains one row per AOI/segment and no analyses span AOI. The AOIs included in this file can come from different GeoMx runs and hence span Globus uploads. So care must be taken to make sure the appropriate AOIs are included in the file. |
| <code>lab_processed\/annotations\.xlsx</code> |  | AOI specific annotations. This might include cell type and anatomical information. |
| <code>lab_processed\/dcc\/.*</code> | ‚úì | DCC files generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/dcc\/[^\/]+\.dcc</code> | ‚úì | DCC files containing target probe counts, generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/primary_analysis\/.*</code> | ‚úì | Primary analysis results |
| <code>lab_processed\/primary_analysis\/[^\/]+\.xlsx</code> | ‚úì | **[QA/QC]** File containing results from initial processing by GeoMx DSP Data Analysis Suite including optional removal of segments and targets based on QC flags and LOQ and normalization using Q3 normalization. |

<summary><b>Version 2.1</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/[^\/]+_LabWorksheet.txt</code> | ‚úì | An Excel spreadsheet to refer to in setting up the library. This file documents all of the samples from a single collection plate. Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_config\.ini</code> | ‚úì | Needed to generate the DCC file from the fastq file. Contains pipeline processing parameters.  Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_SeqCodeIndices\.csv</code> | ‚úì | A file with sample information needed by the Illumina software. Use the contents of the SeqCodeIndices.csv file to create a SampleSheet.csv for input to the Illumina sequencer. (NextSeq 1000/2000 users download a SampleSheet.csv and whitelist.txt instead of SeqCodeIndices.csv.)  Generated by DSP run. |
| <code>raw\/markers\.csv</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection [this should be similar in structure to the antibodies file] |
| <code>raw\/[^\/]*targets\.pkc</code> | ‚úì | The file listing probe barcode sequence and corresponding gene symbol or proteins targeted by that probe. This should be consistent for the same probe panel. |
| <code>raw\/additional_panels_used\.csv</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv</code> | ‚úì | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/fastq\/.*</code> | ‚úì | Raw sequencing files for the experiment |
| <code>raw\/fastq\/oligo\/.*</code> | ‚úì | Directory containing fastq files pertaining to oligo sequencing. |
| <code>raw\/fastq\/oligo\/[^\/]+\.fastq\.gz</code> | ‚úì | This is a gzip version of the fastq file. This file contains the cell barcode and unique molecular identifier (technical). |
| <code>raw\/images\/.*</code> |  | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/Initial\s{1}Dataset\.xlsx</code> | ‚úì | **[QA/QC]** An excel spreadsheet that is downloaded from the GeoMx DSP Data Analysis Suite containing QA/QC metrics based on raw, unprocessed target counts. This file contains one row per AOI/segment and no analyses span AOI. The AOIs included in this file can come from different GeoMx runs and hence span Globus uploads. So care must be taken to make sure the appropriate AOIs are included in the file. |
| <code>lab_processed\/annotations\.xlsx</code> |  | AOI specific annotations. This might include cell type and anatomical information. |
| <code>lab_processed\/dcc\/.*</code> | ‚úì | DCC files generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/dcc\/[^\/]+\.dcc</code> | ‚úì | DCC files containing target probe counts, generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/primary_analysis\/.*</code> | ‚úì | Primary analysis results |
| <code>lab_processed\/primary_analysis\/Q3\s{1}Normalized\.xlsx</code> | ‚úì | **[QA/QC]** Results from initial procesing by GeoMx DSP Data Analysis Suite. The collection of datasets were normalized using Q3 normalization after target genes below the limit of quantitation (LOQ) are removed. |

<summary><b>Version 2.0</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/[^\/]+_LabWorksheet.txt</code> | ‚úì | An Excel spreadsheet to refer to in setting up the library. This file documents all of the samples from a single collection plate. Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_config\.ini</code> | ‚úì | Needed to generate the DCC file from the fastq file. Contains pipeline processing parameters.  Generated by DSP run, prior to sequencing. |
| <code>raw\/[^\/]+_SeqCodeIndices\.csv</code> | ‚úì | A file with sample information needed by the Illumina software. Use the contents of the SeqCodeIndices.csv file to create a SampleSheet.csv for input to the Illumina sequencer. (NextSeq 1000/2000 users download a SampleSheet.csv and whitelist.txt instead of SeqCodeIndices.csv.)  Generated by DSP run. |
| <code>raw\/markers\.csv</code> |  | A csv file describing any morphology markers used to guide ROI and/or AOI selection [this should be similar in structure to the antibodies file] |
| <code>raw\/[^\/]*targets\.pkc</code> | ‚úì | The file listing probe barcode sequence and corresponding gene symbol or proteins targeted by that probe. This should be consistent for the same probe panel. |
| <code>raw\/additional_panels_used\.csv</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv</code> | ‚úì | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/fastq\/.*</code> | ‚úì | Raw sequencing files for the experiment |
| <code>raw\/fastq\/oligo\/.*</code> | ‚úì | Directory containing fastq files pertaining to oligo sequencing. |
| <code>raw\/fastq\/oligo\/[^\/]+\.fastq\.gz</code> | ‚úì | This is a gzip version of the fastq file. This file contains the cell barcode and unique molecular identifier (technical). |
| <code>raw\/images\/.*</code> |  | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/Initial\s{1}Dataset\.xlsx</code> | ‚úì | **[QA/QC]** An excel spreadsheet that is downloaded from the GeoMx DSP Data Analysis Suite containing QA/QC metrics based on raw, unprocessed target counts. This file contains one row per AOI/segment and no analyses span AOI. The AOIs included in this file can come from different GeoMx runs and hence span Globus uploads. So care must be taken to make sure the appropriate AOIs are included in the file. |
| <code>lab_processed\/annotations\.xlsx</code> |  | AOI specific annotations. This might include cell type and anatomical information. |
| <code>lab_processed\/dcc\/.*</code> | ‚úì | DCC files generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/dcc\/[^\/]+\.dcc</code> | ‚úì | DCC files containing target probe counts, generated from fastq by the Nanostring GeoMx NGS Pipeline. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff</code> | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |

