---
title: CosMx
schema_name: cosmx
category: Spatial Transcriptomics
all_versions_deprecated: False
exclude_from_index: False
layout: default

---
Prepare your metadata based on the latest metadata schema using one of the template files below. See the instructions in the [Metadata Validation Workflow](https://docs.google.com/document/d/1lfgiDGbyO4K4Hz1FMsJjmJd9RdwjShtJqFYNwKpbcZY) document for more information on preparing and validating your metadata.tsv file prior to submission.

Related files:


- [üìù Excel template](https://raw.githubusercontent.com/hubmapconsortium/dataset-metadata-spreadsheet/main/cosmx/latest/cosmx.xlsx): For metadata entry.
- [üìù TSV template](https://raw.githubusercontent.com/hubmapconsortium/dataset-metadata-spreadsheet/main/cosmx/latest/cosmx.tsv): Alternative for metadata entry.




## Metadata schema


<summary><a href="https://openview.metadatacenter.org/templates/https:%2F%2Frepo.metadatacenter.org%2Ftemplates%2F3e0aec79-c544-49e2-adf0-9f62fd8546fd"><b>Version 2 (use this one)</b></a></summary>



<br>

## Directory schemas
<summary><b>Version 2.4 (use this one)</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/[^\/]*_exprMat_file\.csv$</code> | ‚úì | Cell expression matrix with raw counts of genes for each identified cell. Apart from the gene specific columns, additional columns include: the origin of the cell (fov, unique cell id); negative probes; probes associated with the fiducial frame (SystemControl) |
| <code>raw\/[^\/]*_fov_positions_file\.csv$</code> | ‚úì | FOV Positions file that provides an overview of the tissue locations and to help specify separate regions and/or tissues on the slide. This contains information about the Slide - slide number it comes from; FOV - field of view; X_mm/Y_mm - x/y coordinates of FOV positions in mm (previously in px) |
| <code>raw\/[^\/]*_metadata_file\.csv$</code> | ‚úì | Cell metadata file containing the following information - the origin of the cell (fov, unique cell id); physical properties of the cell (area, aspect ratio, width, height); location of the cell centroid within each FOV (center X/Y local) and global position (center X/Y global); information about the protein staining (min/max intensity); type of protein, which may be specific to each experiment but generally includes DAPI, Membrane, PanCK, CD45; other (e.g. Seurat information if that pipeline was used within AtoMx, some data quality information) |
| <code>raw\/[^\/]*_tx_file\.csv$</code> | ‚úì | File containing specific transcript location output with information regarding all transcripts detected across various fields of view (FOVs), including cell_ID, global and local coordinates, Z-slice index, target and the subcellular compartment in which the transcript was identified. |
| <code>raw\/markers\.csv$</code> | ‚úì | A csv file describing any morphology markers used to guide ROI and/or AOI selection. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include: manufacturer, model/name, product code. |
| <code>raw\/gene_panel\.csv$</code> | ‚úì | The list of target genes. The expected format is: gene_id (ensembl ID), gene_name. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include: target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/transcript_locations\.csv$</code> |  | Contains description of the location of all decoded transcripts. The origin of the coordinate is 0,0 at the top left corner of the image. The file should include: gene name, x, y, z (optional), quality score (optional). It is expected that the first row in the file contains the column header. |
| <code>raw\/custom_gene_list\.csv$</code> |  | This describes the target genes profiled by the assay. For advanced design, this can be probes sequences for splicing or other analysis for any target of interest. The format should minimally contain: gene name, ensemble ID |
| <code>raw\/probes\.csv$</code> |  | A CSV file describing the probe panel used. This is typically what's used to specifiy the probe set when ordering a probe panel for a Xenium run. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff&#124;tif)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>raw\/images\/preview_scan\.png$</code> | ‚úì | Assists in selection of regions of FOVs using the grid FOV placement tool. |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.(?:tif&#124;tiff)$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-(?:tif&#124;tiff)\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accompanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |

<summary><b>Version 2.3</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/[^\/]*_exprMat_file\.csv$</code> | ‚úì | Cell expression matrix with raw counts of genes for each identified cell. Apart from the gene specific columns, additional columns include: the origin of the cell (fov, unique cell id); negative probes; probes associated with the fiducial frame (SystemControl) |
| <code>raw\/[^\/]*_fov_positions_file\.csv$</code> | ‚úì | FOV Positions file that provides an overview of the tissue locations and to help specify separate regions and/or tissues on the slide. This contains information about the Slide - slide number it comes from; FOV - field of view; X_mm/Y_mm - x/y coordinates of FOV positions in mm (previously in px) |
| <code>raw\/[^\/]*_metadata_file\.csv$</code> | ‚úì | Cell metadata file containing the following information - the origin of the cell (fov, unique cell id); physical properties of the cell (area, aspect ratio, width, height); location of the cell centroid within each FOV (center X/Y local) and global position (center X/Y global); information about the protein staining (min/max intensity); type of protein, which may be specific to each experiment but generally includes DAPI, Membrane, PanCK, CD45; other (e.g. Seurat information if that pipeline was used within AtoMx, some data quality information) |
| <code>raw\/[^\/]*_tx_file\.csv$</code> | ‚úì | File containing specific transcript location output with information regarding all transcripts detected across various fields of view (FOVs), including cell_ID, global and local coordinates, Z-slice index, target and the subcellular compartment in which the transcript was identified. |
| <code>raw\/markers\.csv$</code> | ‚úì | A csv file describing any morphology markers used to guide ROI and/or AOI selection. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include: manufacturer, model/name, product code. |
| <code>raw\/gene_panel\.csv$</code> | ‚úì | The list of target genes. The expected format is: gene_id (ensembl ID), gene_name. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include: target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/transcript_locations\.csv$</code> |  | Contains decription of the location of all decoded transcripts. The origin of the coordinate is 0,0 at the top left corner of the image. The file should include: gene name, x, y, z (optional), quality score (optional). It is expected that the first row in the file contains the column header. |
| <code>raw\/custom_gene_list\.csv$</code> |  | This describes the target genes profiled by the assay. For advanced design, this can be probes sequences for splicing or other analysis for any target of interest. The format should minimally contain: gene name, ensemble ID |
| <code>raw\/probes\.csv$</code> |  | A CSV file describing the probe panel used. This is typically what's used to specifiy the probe set when ordering a probe panel for a Xenium run. |
| <code>raw\/images\/FOV[^\/]*\/.*</code> | ‚úì | Directory containing imaging tiles. |
| <code>raw\/images\/FOV[^\/]*\/[^\/]*_complete_code_cell_target_call_coord\.csv$</code> | ‚úì | Target coordinates and counts per cell. |
| <code>raw\/images\/FOV[^\/]*\/[^\/]+\.tiff$</code> | ‚úì | The tiled image. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>raw\/images\/preview_scan\.png$</code> | ‚úì | Assists in selection of regions of FOVs using the grid FOV placement tool. |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |

<summary><b>Version 2.2</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/[^\/]*_exprMat_file\.csv$</code> | ‚úì | Cell expression matrix with raw counts of genes for each identified cell. Apart from the gene specific columns, additional columns include: the origin of the cell (fov, unique cell id); negative probes; probes associated with the fiducial frame (SystemControl) |
| <code>raw\/[^\/]*_fov_positions_file\.csv$</code> | ‚úì | FOV Positions file that provides an overview of the tissue locations and to help specify separate regions and/or tissues on the slide. This contains information about the Slide - slide number it comes from; FOV - field of view; X_mm/Y_mm - x/y coordinates of FOV positions in mm (previously in px) |
| <code>raw\/[^\/]*_metadata_file\.csv$</code> | ‚úì | Cell metadata file containing the following information - the origin of the cell (fov, unique cell id); physical properties of the cell (area, aspect ratio, width, height); location of the cell centroid within each FOV (center X/Y local) and global position (center X/Y global); information about the protein staining (min/max intensity); type of protein, which may be specific to each experiment but generally includes DAPI, Membrane, PanCK, CD45; other (e.g. Seurat information if that pipeline was used within AtoMx, some data quality information) |
| <code>raw\/[^\/]*_tx_file\.csv$</code> | ‚úì | File containing specific transcript location output with information regarding all transcripts detected across various fields of view (FOVs), including cell_ID, global and local coordinates, Z-slice index, target and the subcellular compartment in which the transcript was identified. |
| <code>raw\/markers\.csv$</code> | ‚úì | A csv file describing any morphology markers used to guide ROI and/or AOI selection. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include: manufacturer, model/name, product code. |
| <code>raw\/gene_panel\.csv$</code> | ‚úì | The list of target genes. The expected format is: gene_id (ensembl ID), gene_name. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include: target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/transcript_locations\.csv$</code> |  | Contains description of the location of all decoded transcripts. The origin of the coordinate is 0,0 at the top left corner of the image. The file should include: gene name, x, y, z (optional), quality score (optional). It is expected that the first row in the file contains the column header. |
| <code>raw\/custom_gene_list\.csv$</code> |  | This describes the target genes profiled by the assay. For advanced design, this can be probes sequences for splicing or other analysis for any target of interest. The format should minimally contain: gene name, ensemble ID |
| <code>raw\/probes\.csv$</code> |  | A CSV file describing the probe panel used. This is typically what's used to specifiy the probe set when ordering a probe panel for a Xenium run. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>raw\/images\/preview_scan\.png$</code> | ‚úì | Assists in selection of regions of FOVs using the grid FOV placement tool. |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accompanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |

<summary><b>Version 2.1</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/[^\/]*_exprMat_file\.csv$</code> | ‚úì | Cell expression matrix with raw counts of genes for each identified cell. Apart from the gene specific columns, additional columns include: the origin of the cell (fov, unique cell id); negative probes; probes associated with the fiducial frame (SystemControl) |
| <code>raw\/[^\/]*_fov_positions_file\.csv$</code> | ‚úì | FOV Positions file that provides an overview of the tissue locations and to help specify separate regions and/or tissues on the slide. This contains information about the Slide - slide number it comes from; FOV - field of view; X_mm/Y_mm - x/y coordinates of FOV positions in mm (previously in px) |
| <code>raw\/[^\/]*_metadata_file\.csv$</code> | ‚úì | Cell metadata file containing the following information - the origin of the cell (fov, unique cell id); physical properties of the cell (area, aspect ratio, width, height); location of the cell centroid within each FOV (center X/Y local) and global position (center X/Y global); information about the protein staining (min/max intensity); type of protein, which may be specific to each experiment but generally includes DAPI, Membrane, PanCK, CD45; other (e.g. Seurat information if that pipeline was used within AtoMx, some data quality information) |
| <code>raw\/[^\/]*_tx_file\.csv$</code> | ‚úì | File containing specific transcript location output with information regarding all transcripts detected across various fields of view (FOVs), including cell_ID, global and local coordinates, Z-slice index, target and the subcellular compartment in which the transcript was identified. |
| <code>raw\/markers\.csv$</code> | ‚úì | A csv file describing any morphology markers used to guide ROI and/or AOI selection. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include: manufacturer, model/name, product code. |
| <code>raw\/gene_panel\.csv$</code> | ‚úì | The list of target genes. The expected format is: gene_id (ensembl ID), gene_name. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include: target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/transcript_locations\.csv$</code> |  | Contains description of the location of all decoded transcripts. The origin of the coordinate is 0,0 at the top left corner of the image. The file should include: gene name, x, y, z (optional), quality score (optional). It is expected that the first row in the file contains the column header. |
| <code>raw\/custom_gene_list\.csv$</code> |  | This describes the target genes profiled by the assay. For advanced design, this can be probes sequences for splicing or other analysis for any target of interest. The format should minimally contain: gene name, ensemble ID |
| <code>raw\/probes\.csv$</code> |  | A CSV file describing the probe panel used. This is typically what's used to specifiy the probe set when ordering a probe panel for a Xenium run. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>raw\/images\/preview_scan\.png$</code> | ‚úì | Assists in selection of regions of FOVs using the grid FOV placement tool. |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accompanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |

<summary><b>Version 2.0</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/[^\/]*_exprMat_file\.csv</code> | ‚úì | Cell expression matrix with raw counts of genes for each identified cell. Apart from the gene specific columns, additional columns include: the origin of the cell (fov, unique cell id); negative probes; probes associated with the fiducial frame (SystemControl) |
| <code>raw\/[^\/]*_fov_positions_file\.csv</code> | ‚úì | FOV Positions file that provides an overview of the tissue locations and to help specify separate regions and/or tissues on the slide. This contains information about the Slide - slide number it comes from; FOV - field of view; X_mm/Y_mm - x/y coordinates of FOV positions in mm (previously in px) |
| <code>raw\/[^\/]*_metadata_file\.csv</code> | ‚úì | Cell metadata file containing the following information - the origin of the cell (fov, unique cell id); physical properties of the cell (area, aspect ratio, width, height); location of the cell centroid within each FOV (center X/Y local) and global position (center X/Y global); information about the protein staining (min/max intensity); type of protein, which may be specific to each experiment but generally includes DAPI, Membrane, PanCK, CD45; other (e.g. Seurat information if that pipeline was used within AtoMx, some data quality information) |
| <code>raw\/[^\/]*_tx_file\.csv</code> | ‚úì | Cell transcript file |
| <code>raw\/[^\/]*_config\.ini</code> | ‚úì | Needed to generate the DCC file from the fastq file. Contains pipeline processing parameters.  Generated by DSP run, prior to sequencing. |
| <code>raw\/markers\.csv</code> | ‚úì | A csv file describing any morphology markers used to guide ROI and/or AOI selection. |
| <code>raw\/additional_panels_used\.csv</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include: manufacturer, model/name, product code. |
| <code>raw\/gene_panel\.csv</code> | ‚úì | The list of target genes. The expected format is: gene_id (ensembl ID), gene_name. |
| <code>raw\/custom_probe_set\.csv</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include: target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/transcript_locations\.csv</code> | ‚úì | The origin of the coordinate is 0,0 at the top left corner of the image. The file should include: gene name, x, y, z (optional), quality score (optional). It is expected that the first row in the file contains the column header. |
| <code>raw\/custom_gene_list\.csv</code> |  | This describes the target genes profiled by the assay. For advanced design, this can be probes sequences for splicing or other analysis for any target of interest. The format should minimally contain: gene name, ensemble ID |
| <code>raw\/probes\.csv</code> |  | A CSV file describing the probe panel used. This is typically what's used to specifiy the probe set when ordering a probe panel for a Xenium run. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>raw\/images\/preview_scan\.png</code> | ‚úì | Assists in selection of regions of FOVs using the grid FOV placement tool. |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |

