---
title: CosMx Proteomics
schema_name: cosmx-proteomics
category: Proteomics
all_versions_deprecated: False
exclude_from_index: False
layout: default

---
Prepare your metadata based on the latest metadata schema using one of the template files below. See the instructions in the [Metadata Validation Workflow](https://docs.google.com/document/d/1lfgiDGbyO4K4Hz1FMsJjmJd9RdwjShtJqFYNwKpbcZY) document for more information on preparing and validating your metadata.tsv file prior to submission.

Related files:


- [üìù Excel template](https://raw.githubusercontent.com/hubmapconsortium/dataset-metadata-spreadsheet/main/cosmx-proteomics/latest/cosmx-proteomics.xlsx): For metadata entry.
- [üìù TSV template](https://raw.githubusercontent.com/hubmapconsortium/dataset-metadata-spreadsheet/main/cosmx-proteomics/latest/cosmx-proteomics.tsv): Alternative for metadata entry.




## Metadata schema


<summary><a href="https://openview.metadatacenter.org/templates/https:%2F%2Frepo.metadatacenter.org%2Ftemplates%2F971357a5-6491-4149-aa6e-d6f14bc9a048"><b>Version 2 (use this one)</b></a></summary>



<br>

## Directory schemas
<summary><b>Version 2.1 (use this one)</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/[^\/]*_exprMat_file\.csv$</code> | ‚úì | Cell expression matrix with raw counts of genes for each identified cell. Apart from the gene specific columns, additional columns include: the origin of the cell (fov, unique cell id); negative probes; probes associated with the fiducial frame (SystemControl) |
| <code>raw\/[^\/]*_fov_positions_file\.csv$</code> | ‚úì | FOV Positions file that provides an overview of the tissue locations and to help specify separate regions and/or tissues on the slide. This contains information about the Slide - slide number it comes from; FOV - field of view; X_mm/Y_mm - x/y coordinates of FOV positions in mm (previously in px) |
| <code>raw\/latest\.fovs\.csv$</code> |  | Legacy version of file located in the RunSummary directory of AtoMx exports that is distinct from the *fov_positions_file.csv and include Z-axis and Z-offset information. |
| <code>raw\/FOV_locations\.csv$</code> |  | Current version of file located in the RunSummary directory of AtoMx exports that is distinct from the *fov_positions_file.csv and include Z-axis and Z-offset information. |
| <code>raw\/[^\/]*_metadata_file\.csv$</code> | ‚úì | Cell metadata file containing the following information - the origin of the cell (fov, unique cell id); physical properties of the cell (area, aspect ratio, width, height); location of the cell centroid within each FOV (center X/Y local) and global position (center X/Y global); information about the protein staining (min/max intensity); type of protein, which may be specific to each experiment but generally includes DAPI, Membrane, PanCK, CD45; other (e.g. Seurat information if that pipeline was used within AtoMx, some data quality information) |
| <code>raw\/[^\/]*polygons\.csv$</code> | ‚úì | File containing the coordinate data for the polygons outlining the boundaries of every cell identified during the imaging process. |
| <code>raw\/plex[^\/]*\.txt$</code> | ‚úì | File containing ProbeID-to-DisplayName keys. |
| <code>raw\/markers\.csv$</code> | ‚úì | A csv file describing any morphology markers used to guide ROI and/or AOI selection. This should minimally contain name of fluorophor, vendor, channel. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include: manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include: target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/images\/FOV[^\/]*\/.*</code> | ‚úì | Directory containing imaging tiles. |
| <code>raw\/images\/FOV[^\/]*\/[^\/]+\.tiff$</code> | ‚úì | The tiled image. |
| <code>raw\/images\/FOV[^\/]*\/[^\/]+\.tiff$</code> | ‚úì | The tiled image. |
| <code>raw\/images\/FOV[^\/]*\/ProteinImages\/[^\/]+\.tiff$</code> | ‚úì | High-resolution, multi-layered images generated by the CosMx Spatial Molecular Imager (SMI) that contain the spatial data for up to 76 different proteins imaged at single-cell and subcellular resolution within a tissue sample |
| <code>raw\/images\/FOV[^\/]*\/ProteinMasks\/[^\/]+\.tiff$</code> | ‚úì | High-resolution image file that shows the segmentation of cells within a tissue sample, based on the protein staining data generated by the Bruker CosMx Spatial Molecular Imager. This mask file outlines the boundaries of individual cells and is used to accurately identify and analyze protein expression at the single-cell level. |
| <code>raw\/images\/FOV[^\/]*\/overlay\.(?:jpeg&#124;tif&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>raw\/images\/FOV[^\/]*\/CellOverlay_FOV[^\/]*\.jpg$</code> |  | File depicting a visual overlay of key spatial transcriptomics data for the specific field of view (FOV). These images combine cellular morphology with transcript information to provide context for the gene expression data. |
| <code>raw\/images\/FOV[^\/]*\/CellLabels_F[^\/]*\.(?:tif&#124;tiff)$</code> |  | A TIFF or TIF image containing the results of cell segmentation. It provides a visual map of the identified cells, with each cell in a field of view (FOV) assigned a unique numerical label. |
| <code>raw\/images\/preview_scan\.png$</code> | ‚úì | Assists in selection of regions of FOVs using the grid FOV placement tool. |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> |  | Processed image files |

<summary><b>Version 2.0</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/[^\/]*_exprMat_file\.csv$</code> | ‚úì | Cell expression matrix with raw counts of genes for each identified cell. Apart from the gene specific columns, additional columns include: the origin of the cell (fov, unique cell id); negative probes; probes associated with the fiducial frame (SystemControl) |
| <code>raw\/[^\/]*_fov_positions_file\.csv$</code> | ‚úì | FOV Positions file that provides an overview of the tissue locations and to help specify separate regions and/or tissues on the slide. This contains information about the Slide - slide number it comes from; FOV - field of view; X_mm/Y_mm - x/y coordinates of FOV positions in mm (previously in px) |
| <code>raw\/[^\/]*_metadata_file\.csv$</code> | ‚úì | Cell metadata file containing the following information - the origin of the cell (fov, unique cell id); physical properties of the cell (area, aspect ratio, width, height); location of the cell centroid within each FOV (center X/Y local) and global position (center X/Y global); information about the protein staining (min/max intensity); type of protein, which may be specific to each experiment but generally includes DAPI, Membrane, PanCK, CD45; other (e.g. Seurat information if that pipeline was used within AtoMx, some data quality information) |
| <code>raw\/markers\.csv$</code> | ‚úì | A csv file describing any morphology markers used to guide ROI and/or AOI selection. This should minimally contain name of fluorophor, vendor, channel. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include: manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include: target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/images\/FOV[^\/]*\/.*</code> | ‚úì | Directory containing imaging tiles. |
| <code>raw\/images\/FOV[^\/]*\/[^\/]+\.tiff$</code> | ‚úì | The tiled image. |
| <code>raw\/images\/overlay\.(?:jpeg&#124;tiff)$</code> |  | State whether an overlay image was used to guide ROI selection. If an overlay is used, then the overlay details will be provided in the protocols.io protocol. If used, this needs to be uploaded. It is not included in the OME TIFF. This can be a JPEG or TIFF file |
| <code>raw\/images\/preview_scan\.png$</code> | ‚úì | Assists in selection of regions of FOVs using the grid FOV placement tool. |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> |  | Processed image files |
| <code>lab_processed\/images\/[^\/]+tissue-boundary\.geojson</code> |  | **[QA/QC]** If the boundaries of the tissue have been identified (e.g., by manual efforts), then the boundary geometry can be included as a GeoJSON file named ‚Äú*.tissue-boundary.geojson‚Äù. |

