---
title: MERFISH
schema_name: merfish
category: Fluorescence In Situ Hybridization (FISH)
all_versions_deprecated: False
exclude_from_index: False
layout: default

---
Prepare your metadata based on the latest metadata schema using one of the template files below. See the instructions in the [Metadata Validation Workflow](https://docs.google.com/document/d/1lfgiDGbyO4K4Hz1FMsJjmJd9RdwjShtJqFYNwKpbcZY) document for more information on preparing and validating your metadata.tsv file prior to submission.

Related files:


- [üìù Excel template](https://raw.githubusercontent.com/hubmapconsortium/dataset-metadata-spreadsheet/main/merfish/latest/merfish.xlsx): For metadata entry.
- [üìù TSV template](https://raw.githubusercontent.com/hubmapconsortium/dataset-metadata-spreadsheet/main/merfish/latest/merfish.tsv): Alternative for metadata entry.


[This link](https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0) lists the set of fields that are required in the OME TIFF file XML header.

## Metadata schema


<summary><a href="https://openview.metadatacenter.org/templates/https:%2F%2Frepo.metadatacenter.org%2Ftemplates%2Ff1ef260f-d4a3-43db-a739-49b394aeee20"><b>Version 2 (use this one)</b></a></summary>



<br>

## Directory schemas
<summary><b>Version 2.8 (use this one)</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include: manufacturer, model/name, product code. |
| <code>raw\/gene_panel\.csv$</code> | ‚úì | The list of target genes. The expected format is: gene_id (ensembl ID), gene_name. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include: target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file> ). |
| <code>raw\/micron_to_mosaic_pixel_transform\.csv$</code> |  | Matrix used to transform from pixels to physical distance. |
| <code>raw\/manifest\.json$</code> | ‚úì | This file contains stain by channel details and pixel details. |
| <code>raw\/codebook\.csv$</code> | ‚úì | CSV containing codebook information for the experiment. Rows are barcodes and columns are imaging rounds. The first column is the barcode target, and the following column IDs are expected to be sequential, and round identifiers are expected to be integers (not roman numerals). |
| <code>raw\/positions\.csv$</code> | ‚úì | File that includes the top left coordinate of each tiled image. This is required to stitch the images. |
| <code>raw\/dataorganization\.csv$</code> | ‚úì | Necessary image definitions. |
| <code>raw\/data\/.*</code> |  | All raw stack data files for the MERFISH experiment. |
| <code>raw\/data\/[^\/]+\.dax$</code> |  | The raw image stack for MERFISH v1. |
| <code>raw\/data\/[^\/]+\.inf$</code> |  | Information file with dax image format specifications for MERFISH v1. Variable expected for downstream processing with PIPEFISH are: frame dimensions, number of frames, little/big endian, stage X and Y locations, lock target, scalemin, and scalemax. |
| <code>raw\/data\/[^\/]+\.zip$</code> |  | The raw image stack for MERFISH v2. |
| <code>raw\/images\/.*</code> | ‚úì | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/[^\/]+\.tif$</code> |  | Raw microscope file for the experiment in MERFISH v.1. |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files. |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) |  | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> |  | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*tissue-boundary\.geojson$</code> |  | **[QA/QC]** If the boundaries of the tissue have been identified (e.g., by manual efforts), then the boundary geometry can be included as a GeoJSON file named "*tissue-boundary.geojson". |
| <code>lab_processed\/merfish_output\/.*</code> | ‚úì | Directory containing the MERFISH cell x gene analysis. |
| <code>lab_processed\/merfish_output\/detected_transcripts\.csv$</code> | ‚úì | File containing a list of detected transcripts in sample. Transcripts are stored as a list with the following associated metadata. barcode_id, global_x, global_y, global_z, x, y, fov, transcript_score, gene, transcript_id, cell_id. |
| <code>lab_processed\/merfish_output\/detected_transcripts\.parquet$</code> |  | File containing a list of detected transcripts in sample. Transcripts are stored as a list with x, y, and z locations in the sample. |
| <code>lab_processed\/merfish_output\/images\/.*</code> |  | Directory containing the MERFISH mosaic image files. |
| <code>lab_processed\/merfish_output\/images\/mosaic_\.[^\/]+\.tif$</code> |  | File containing a stitched mosaic of whole sample. Mosaic tif files are downsampled from high resolution raw imaging data. |
| <code>lab_processed\/merfish_output\/cell_boundaries\.parquet$</code> |  | File containing a list of cell boundary coordinates. |
| <code>lab_processed\/merfish_output\/cell_by_gene\.csv$</code> |  | File containing a table of cells with associated number of transcripts measured. |
| <code>lab_processed\/merfish_output\/cell_categories\.csv$</code> |  | File containing annotated cell types found in sample. |
| <code>lab_processed\/merfish_output\/cell_metadata\.csv$</code> |  | File containing additional cell metadata. |
| <code>lab_processed\/merfish_output\/differentially_expressed_genes\.csv$</code> |  | File containing a list of genes calculated to be differentially expressed in a thresholded population of cells in the sample. |
| <code>lab_processed\/merfish_output\/summary\.png$</code> |  | The summary.png file is a visual summary of the various QC metrics for the experiment, including transcript detection, segmentation, etc. |
| <code>lab_processed\/merfish_output\/[^\/]+\.h5ad$</code> | ‚úì | Anndata formated collation of cell_by_gene.csv, cell_metadata.csv, as well as some additonal QC metrics and leiden categories. |
| <code>lab_processed\/merfish_output\/[^\/]+\.vzg2$</code> |  | Vizgen specific file for use in loading experiment into the MERSCOPE Vizualizer software. |
| <code>lab_processed\/merfish_output\/report_[^\/]+\.html$</code> |  | Interactive report of wide array of QC metrics. |
| <code>lab_processed\/merfish_output\/dapi\.png$</code> | ‚úì | File containing a stitched mosaic of dapi stain for whole sample. |

<summary><b>Version 2.7</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/gene_panel\.csv$</code> | ‚úì | The list of target genes. The expected format is gene_id (ensembl ID), gene_name. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/micron_to_mosaic_pixel_transform\.csv$</code> |  | Matrix used to transform from pixels to physical distance. |
| <code>raw\/manifest\.json$</code> | ‚úì | This file contains stain by channel details and pixel details. |
| <code>raw\/codebook\.csv$</code> | ‚úì | CSV containing codebook information for the experiment. Rows are barcodes and columns are imaging rounds. The first column is the barcode target, and the following column IDs are expected to be sequential, and round identifiers are expected to be integers (not roman numerals). |
| <code>raw\/positions\.csv$</code> | ‚úì | File that includes the top left coordinate of each tiled image. This is required to stitch the images. |
| <code>raw\/dataorganization\.csv$</code> | ‚úì | Necessary image definitions |
| <code>raw\/data\/.*</code> | ‚úì | All raw stack data files for the MERFISH experiment. |
| <code>raw\/data\/[^\/]+\.dax$</code> | ‚úì | The raw image stack. |
| <code>raw\/data\/[^\/]+\.inf$</code> | ‚úì | Information file with dax image format specifications. Variable expected for downstream processing with PIPEFISH are frame dimensions, number of frames, little/big endian, stage X and Y locations, lock target, scalemin, and scalemax. |
| <code>raw\/images\/.*</code> | ‚úì | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/[^\/]+\.tif$</code> | ‚úì | Raw microscope file for the experiment. |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/detected_transcripts\.csv$</code> | ‚úì | A file containing the locations of each RNA target. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*\.tissue-boundary\.geojson$</code> |  | **[QA/QC]** If the boundaries of the tissue have been identified (e.g., by manual efforts), then the boundary geometry can be included as a GeoJSON file named ‚Äú*.tissue-boundary.geojson‚Äù. |

<summary><b>Version 2.6</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/gene_panel\.csv$</code> | ‚úì | The list of target genes. The expected format is gene_id (ensembl ID), gene_name. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/micron_to_mosaic_pixel_transform\.csv$</code> |  | Matrix used to transform from pixels to physical distance. |
| <code>raw\/manifest\.json$</code> | ‚úì | This file contains stain by channel details and pixel details. |
| <code>raw\/codebook\.csv$</code> | ‚úì | CSV containing codebook information for the experiment. Rows are barcodes and columns are imaging rounds. The first column is the barcode target, and the following column IDs are expected to be sequential, and round identifiers are expected to be integers (not roman numerals). |
| <code>raw\/positions\.csv$</code> | ‚úì | File that includes the top left coordinate of each tiled image. This is required to stitch the images. |
| <code>raw\/dataorganization\.csv$</code> | ‚úì | Necessary image definitions |
| <code>raw\/data\/.*</code> | ‚úì | All raw stack data files for the MERFISH experiment. |
| <code>raw\/data\/[^\/]+\.dax$</code> | ‚úì | The raw image stack. |
| <code>raw\/data\/[^\/]+\.inf$</code> | ‚úì | Information file with dax image format specifications. Variable expected for downstream processing with PIPEFISH are frame dimensions, number of frames, little/big endian, stage X and Y locations, lock target, scalemin, and scalemax. |
| <code>raw\/images\/.*</code> | ‚úì | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/[^\/]+\.tif$</code> | ‚úì | Raw microscope file for the experiment. |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/detected_transcripts\.csv$</code> | ‚úì | A file containing the locations of each RNA target. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |

<summary><b>Version 2.5</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/gene_panel\.csv$</code> | ‚úì | The list of target genes. The expected format is gene_id (ensembl ID), gene_name. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/micron_to_mosaic_pixel_transform\.csv$</code> |  | Matrix used to transform from pixels to physical distance. |
| <code>raw\/manifest\.json$</code> | ‚úì | This file contains stain by channel details and pixel details. |
| <code>raw\/codebook\.csv$</code> | ‚úì | CSV containing codebook information for the experiment. Rows are barcodes and columns are imaging rounds. The first column is the barcode target, and the following column IDs are expected to be sequential, and round identifiers are expected to be integers (not roman numerals). |
| <code>raw\/positions\.csv$</code> | ‚úì | File that includes the top left coordinate of each tiled image. This is required to stitch the images. |
| <code>raw\/dataorganization\.csv$</code> | ‚úì | Necessary image definitions |
| <code>raw\/data\/.*</code> | ‚úì | All raw stack data files for the MERFISH experiment. |
| <code>raw\/data\/[^\/]+\.dax$</code> | ‚úì | The raw image stack. |
| <code>raw\/data\/[^\/]+\.inf$</code> | ‚úì | Information file with dax image format specifications. Variable expected for downstream processing with PIPEFISH are frame dimensions, number of frames, little/big endian, stage X and Y locations, lock target, scalemin, and scalemax. |
| <code>raw\/images\/.*</code> | ‚úì | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/[^\/]+\.(?:tif&#124;tiff)$</code> | ‚úì | Raw microscope file for the experiment. |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/detected_transcripts\.csv$</code> | ‚úì | A file containing the locations of each RNA target. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.(?:tif&#124;tiff)$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-(?:tif&#124;tiff)\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |

<summary><b>Version 2.4</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/gene_panel\.csv$</code> | ‚úì | The list of target genes. The expected format is gene_id (ensembl ID), gene_name. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/micron_to_mosaic_pixel_transform\.csv$</code> |  | Matrix used to transform from pixels to physical distance. |
| <code>raw\/manifest\.json$</code> | ‚úì | This file contains stain by channel details and pixel details. |
| <code>raw\/codebook\.csv$</code> | ‚úì | CSV containing codebook information for the experiment. Rows are barcodes and columns are imaging rounds. The first column is the barcode target, and the following column IDs are expected to be sequential, and round identifiers are expected to be integers (not roman numerals). |
| <code>raw\/positions\.csv$</code> | ‚úì | File that includes the top left coordinate of each tiled image. This is required to stitch the images. |
| <code>raw\/dataorganization\.csv$</code> | ‚úì | Necessary image definitions |
| <code>raw\/data\/.*</code> | ‚úì | All raw stack data files for the MERFISH experiment. |
| <code>raw\/data\/[^\/]+\.dax$</code> | ‚úì | The raw image stack. |
| <code>raw\/data\/[^\/]+\.inf$</code> | ‚úì | Information file with dax image format specifications. Variable expected for downstream processing with PIPEFISH are frame dimensions, number of frames, little/big endian, stage X and Y locations, lock target, scalemin, and scalemax. |
| <code>raw\/images\/.*</code> | ‚úì | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/[^\/]+\.tif$</code> | ‚úì | Raw microscope file for the experiment. |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/detected_transcripts\.csv$</code> | ‚úì | A file containing the locations of each RNA target. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv$</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |

<summary><b>Version 2.3</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/additional_panels_used\.csv</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/gene_panel\.csv</code> | ‚úì | The list of target genes. The expected format is gene_id (ensembl ID), gene_name. |
| <code>raw\/custom_probe_set\.csv</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/micron_to_mosaic_pixel_transform\.csv</code> |  | Matrix used to transform from pixels to physical distance. |
| <code>raw\/manifest\.json</code> | ‚úì | This file contains stain by channel details and pixel details. |
| <code>raw\/codebook\.csv</code> | ‚úì | CSV containing codebook information for the experiment. Rows are barcodes and columns are imaging rounds. The first column is the barcode target, and the following column IDs are expected to be sequential, and round identifiers are expected to be integers (not roman numerals). |
| <code>raw\/positions\.csv</code> | ‚úì | File that includes the top left coordinate of each tiled image. This is required to stitch the images. |
| <code>raw\/dataorganization\.csv</code> | ‚úì | Necessary image definitions |
| <code>raw\/data\/.*</code> | ‚úì | All raw stack data files for the MERFISH experiment. |
| <code>raw\/data\/[^\/]+\.dax</code> | ‚úì | The raw image stack. |
| <code>raw\/data\/[^\/]+\.inf</code> | ‚úì | Information file with dax image format specifications. Variable expected for downstream processing with PIPEFISH are frame dimensions, number of frames, little/big endian, stage X and Y locations, lock target, scalemin, and scalemax. |
| <code>raw\/images\/.*</code> | ‚úì | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/[^\/]+\.tif</code> | ‚úì | Raw microscope file for the experiment. |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/detected_transcripts\.csv</code> | ‚úì | A file containing the locations of each RNA target. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |

<summary><b>Version 2.2</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/additional_panels_used\.csv</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/gene_panel\.csv</code> | ‚úì | The list of target genes. The expected format is gene_id (ensembl ID), gene_name. |
| <code>raw\/custom_probe_set\.csv</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/micron_to_mosaic_pixel_transform\.csv</code> |  | Matrix used to transform from pixels to physical distance. |
| <code>raw\/manifest\.json</code> | ‚úì | This file contains stain by channel details and pixel details. |
| <code>raw\/codebook\.csv</code> | ‚úì | CSV containing codebook information for the experiment. Rows are barcodes and columns are imaging rounds. The first column is the barcode target, and the following column IDs are expected to be sequential, and round identifiers are expected to be integers (not roman numerals). |
| <code>raw\/positions\.csv</code> | ‚úì | File that includes the top left coordinate of each tiled image. This is required to stitch the images. |
| <code>raw\/dataorganization\.csv</code> | ‚úì | Necessary image definitions |
| <code>raw\/[^\/]+\.DAX</code> | ‚úì | The raw image stack. |
| <code>raw\/images\/.*</code> | ‚úì | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/[^\/]+\.tif</code> | ‚úì | Raw microscope file for the experiment. |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/detected_transcripts\.csv</code> | ‚úì | A file containing the locations of each RNA target. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |

<summary><b>Version 2.1</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/additional_panels_used\.csv</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/gene_panel\.csv</code> | ‚úì | The list of target genes. |
| <code>raw\/custom_probe_set\.csv</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/micron_to_mosaic_pixel_transform\.csv</code> |  | Matrix used to transform from pixels to physical distance. |
| <code>raw\/manifest\.json</code> | ‚úì | This file contains stain by channel details and pixel details. |
| <code>raw\/codebook\.csv</code> | ‚úì | CSV containing codebook information for the experiment. Rows are barcodes and columns are imaging rounds. The first column is the barcode target, and the following column IDs are expected to be sequential, and round identifiers are expected to be integers (not roman numerals). |
| <code>raw\/positions\.csv</code> | ‚úì | File that includes the top left coordinate of each tiled image. This is required to stitch the images. |
| <code>raw\/dataorganization\.csv</code> | ‚úì | Necessary image definitions |
| <code>raw\/data\/.*</code> | ‚úì | All raw stack data files for the MERFISH experiment. |
| <code>raw\/data\/[^\/]+\.dax</code> | ‚úì | The raw image stack. |
| <code>raw\/data\/[^\/]+\.inf</code> | ‚úì | Information file with dax image format specifications. Variable expected for downstream processing with PIPEFISH are frame dimensions, number of frames, little/big endian, stage X and Y locations, lock target, scalemin, and scalemax. |
| <code>raw\/images\/.*</code> | ‚úì | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/[^\/]+\.tif</code> | ‚úì | Raw microscope file for the experiment. |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/detected_transcripts\.csv</code> | ‚úì | A file containing the locations of each RNA target. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |

<summary><b>Version 2.0</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/microscope_hardware\.json</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/additional_panels_used\.csv</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/gene_panel\.csv</code> | ‚úì | The list of target genes. |
| <code>raw\/custom_probe_set\.csv</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/micron_to_mosaic_pixel_transform\.csv</code> |  | Matrix used to transform from pixels to physical distance. |
| <code>raw\/manifest\.json</code> | ‚úì | This file contains stain by channel details and pixel details. |
| <code>raw\/codebook\.csv</code> | ‚úì | CSV containing codebook information for the experiment. Rows are barcodes and columns are imaging rounds. The first column is the barcode target, and the following column IDs are expected to be sequential, and round identifiers are expected to be integers (not roman numerals). |
| <code>raw\/positions\.csv</code> | ‚úì | File that includes the top left coordinate of each tiled image. This is required to stitch the images. |
| <code>raw\/dataorganization\.csv</code> | ‚úì | Necessary image definitions |
| <code>raw\/[^\/]+\.DAX</code> | ‚úì | The raw image stack. |
| <code>raw\/images\/.*</code> | ‚úì | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/[^\/]+\.tif</code> | ‚úì | Raw microscope file for the experiment. |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/detected_transcripts\.csv</code> | ‚úì | A file containing the locations of each RNA target. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |

