# bulkatacseq

Related files:
- [üî¨ Background doc](TODO): More details about this type.
- [üìù TSV template](https://raw.githubusercontent.com/hubmapconsortium/ingest-validation-tools/master/docs/bulkatacseq/template.tsv): Use this to submit metadata.
- [üíª Source code](https://github.com/hubmapconsortium/ingest-validation-tools/edit/master/src/ingest_validation_tools/table-schemas/level-2/bulkatacseq.yaml): Make a PR if this doc should be updated.

## Table of contents
<details><summary>Provenance</summary>

[`donor_id`](#donor_id)<br>
[`tissue_id`](#tissue_id)<br>
</details>

<details><summary>Level 1</summary>

[`execution_datetime`](#execution_datetime)<br>
[`protocols_io_doi`](#protocols_io_doi)<br>
[`operator`](#operator)<br>
[`operator_email`](#operator_email)<br>
[`pi`](#pi)<br>
[`pi_email`](#pi_email)<br>
[`assay_category`](#assay_category)<br>
[`assay_type`](#assay_type)<br>
[`analyte_class`](#analyte_class)<br>
[`is_targeted`](#is_targeted)<br>
</details>

<details><summary>Level 2</summary>

[`acquisition_instrument_vendor`](#acquisition_instrument_vendor)<br>
[`acquisition_instrument_model`](#acquisition_instrument_model)<br>
[`bulk_transposition_input_number_nuclei`](#bulk_transposition_input_number_nuclei)<br>
[`bulk_atac_cell_isolation_protocols_io_doi`](#bulk_atac_cell_isolation_protocols_io_doi)<br>
[`is_technical_replicate`](#is_technical_replicate)<br>
[`library_adapter_sequence`](#library_adapter_sequence)<br>
[`library_average_fragment_size`](#library_average_fragment_size)<br>
[`library_concentration_value`](#library_concentration_value)<br>
[`library_concentration_unit`](#library_concentration_unit)<br>
[`library_construction_protocols_io_doi`](#library_construction_protocols_io_doi)<br>
[`library_creation_date`](#library_creation_date)<br>
[`library_final_yield_value`](#library_final_yield_value)<br>
[`library_final_yield_unit`](#library_final_yield_unit)<br>
[`library_id`](#library_id)<br>
[`library_layout`](#library_layout)<br>
[`library_pcr_cycles`](#library_pcr_cycles)<br>
[`library_preparation_kit`](#library_preparation_kit)<br>
[`sample_quality_metric`](#sample_quality_metric)<br>
[`sequencing_phix_percent`](#sequencing_phix_percent)<br>
[`sequencing_read_format`](#sequencing_read_format)<br>
[`sequencing_read_percent_q30`](#sequencing_read_percent_q30)<br>
[`sequencing_reagent_kit`](#sequencing_reagent_kit)<br>
[`transposition_kit_number`](#transposition_kit_number)<br>
[`transposition_method`](#transposition_method)<br>
[`transposition_transposase_source`](#transposition_transposase_source)<br>
</details>

<details><summary>Paths</summary>

[`metadata_path`](#metadata_path)<br>
[`data_path`](#data_path)<br></details>

## Provenance

### `donor_id`
HuBMAP Display ID of the donor of the assayed tissue.

| constraint | value |
| --- | --- |
| pattern | `[A-Z]+[0-9]+` |
| required | `True` |

### `tissue_id`
HuBMAP Display ID of the assayed tissue.

| constraint | value |
| --- | --- |
| pattern | `([A-Z]+[0-9]+)-(BL|BR|LB|RB|HT|LK|RK|LI|LV|LL|RL|LY\d\d|SI|SP|TH|TR|UR|OT)(-\d+)+(_\d+)?` |
| required | `True` |

## Level 1

### `execution_datetime`
Start date and time of assay, typically a date-time stamped folder generated by the acquisition instrument. YYYY-MM-DD hh:mm, where YYYY is the year, MM is the month with leading 0s, and DD is the day with leading 0s, hh is the hour with leading zeros, mm are the minutes with leading zeros.

| constraint | value |
| --- | --- |
| type | `datetime` |
| format | `%Y-%m-%d %H:%M` |
| required | `True` |

### `protocols_io_doi`
DOI for protocols.io referring to the protocol for this assay.

| constraint | value |
| --- | --- |
| required | `True` |
| pattern | `10\.17504/.*` |

### `operator`
Name of the person responsible for executing the assay.

| constraint | value |
| --- | --- |
| required | `True` |

### `operator_email`
Email address for the operator.

| constraint | value |
| --- | --- |
| format | `email` |
| required | `True` |

### `pi`
Name of the principal investigator responsible for the data.

| constraint | value |
| --- | --- |
| required | `True` |

### `pi_email`
Email address for the principal investigator.

| constraint | value |
| --- | --- |
| format | `email` |
| required | `True` |

### `assay_category`
Each assay is placed into one of the following 3 general categories: generation of images of microscopic entities, identification & quantitation of molecules by mass spectrometry, and determination of nucleotide sequence.

| constraint | value |
| --- | --- |
| enum | `imaging`, `mass_spectrometry`, or `sequence` |
| required | `True` |

### `assay_type`
The specific type of assay being executed.

| constraint | value |
| --- | --- |
| enum | `scRNA-Seq (10xGenomics)`, `AF`, `bulk RNA`, `bulkATACseq`, `CODEX`, `Imaging Mass Cytometry`, `LC-MS (metabolomics)`, `LC-MS/MS (label-free proteomics)`, `MxIF`, `IMS positive`, `IMS negative`, `MS (shotgun lipidomics)`, `PAS microscopy`, `scATACseq`, `sciATACseq`, `sciRNAseq`, `seqFISH`, `SNARE-seq2`, `snATACseq`, `snRNA`, `SPLiT-Seq`, `TMT (proteomics)`, or `WGS` |
| required | `True` |

### `analyte_class`
Analytes are the target molecules being measured with the assay.

| constraint | value |
| --- | --- |
| enum | `DNA`, `RNA`, `protein`, `lipids`, `metabolites`, or `polysaccharides` |
| required | `True` |

### `is_targeted`
Specifies whether or not a specific molecule(s) is/are targeted for detection/measurement by the assay. The CODEX analyte is protein.

| constraint | value |
| --- | --- |
| type | `boolean` |
| required | `True` |

## Level 2

### `acquisition_instrument_vendor`
An acquisition_instrument is the device that contains the signal detection hardware and signal processing software. Assays generate signals such as light of various intensities or color or signals representing molecular mass.

| constraint | value |
| --- | --- |
| required | `True` |

### `acquisition_instrument_model`
Manufacturers of an acquisition instrument may offer various versions (models) of that instrument with different features or sensitivities. Differences in features or sensitivities may be relevant to processing or interpretation of the data

| constraint | value |
| --- | --- |
| required | `True` |

### `bulk_transposition_input_number_nuclei`
A number (no comma separators)

| constraint | value |
| --- | --- |
| required | `True` |

### `bulk_atac_cell_isolation_protocols_io_doi`
Link to a protocols document answering the question: How was tissue stored and processed for cell/nuclei isolation

| constraint | value |
| --- | --- |
| required | `True` |
| pattern | `10\.17504/.*` |

### `is_technical_replicate`
Is this a sequencing replicate?

| constraint | value |
| --- | --- |
| type | `boolean` |
| required | `True` |

### `library_adapter_sequence`
Adapter sequence to be used for adapter trimming

| constraint | value |
| --- | --- |
| required | `True` |

### `library_average_fragment_size`
Average size in bp of sequencing library fragments estimated via gel electrophoresis or bioanalyzer/tapestation. TODO - Unit field? Ask Nils.

| constraint | value |
| --- | --- |
| required | `True` |

### `library_concentration_value`
The concentration of the library which was submitted for sequencing. This field is different from the "library_final_yield" because the concentration field takes into account how the library was pooled with other libraries. The "library_final_yield" should be greater than or equal to this column. - TODO - Is this correct? This is a concentration, the other is a yield, typically in ng, so they don't seem to have the same units, so an inequality makes no sense.

| constraint | value |
| --- | --- |
| type | `number` |
| required | `True` |

### `library_concentration_unit`
Unit of library_concentration_value

| constraint | value |
| --- | --- |
| enum | `nM` |
| required | `True` |

### `library_construction_protocols_io_doi`
A link to the protocol document containing the library construction method (including version) that was used, e.g. "Smart-Seq2", "Drop-Seq", "10X v3".

| constraint | value |
| --- | --- |
| required | `True` |
| pattern | `10\.17504/.*` |

### `library_creation_date`
date and time of library creation. YYYY-MM-DD, where YYYY is the year, MM is the month with leading 0s, and DD is the day with leading 0s - TODO - constraint. Leave blank if not applicable.

| constraint | value |
| --- | --- |
| required | `False` |

### `library_final_yield_value`
Total amount (eg. nanograms) of library after the clean-up step of final pcr amplification step. Answer the question: What is the Qubit measured concentration (ng/ul) times the elution volume (ul) after the final clean-up step?

| constraint | value |
| --- | --- |
| type | `number` |
| required | `True` |

### `library_final_yield_unit`
Units of library final yield

| constraint | value |
| --- | --- |
| enum | `ng` |
| required | `True` |

### `library_id`
TODO

| constraint | value |
| --- | --- |
| required | `True` |

### `library_layout`
State whether the library was generated for single-end or paired end sequencing. TODO - enum

| constraint | value |
| --- | --- |
| required | `True` |

### `library_pcr_cycles`
Number of PCR cycles performed in order to add adapters and amplify the library. Usually, this includes 5 pre-amplificationn cycles followed by 0-5 additional cycles determined by qPCR.

| constraint | value |
| --- | --- |
| type | `integer` |
| required | `True` |

### `library_preparation_kit`
Reagent kit used for library preparation

| constraint | value |
| --- | --- |
| required | `True` |

### `sample_quality_metric`
This is a quality metric by visual inspection. This should answer the question: Are the nuclei intact and are the nuclei free of significant amounts of debris? This can be captured at a high level, ‚ÄúOK‚Äù or ‚Äúnot OK‚Äù.

| constraint | value |
| --- | --- |
| required | `True` |

### `sequencing_phix_percent`
Percent PhiX loaded to the run

| constraint | value |
| --- | --- |
| type | `number` |
| required | `True` |
| minimum | `0` |
| maximum | `100` |

### `sequencing_read_format`
Number of sequencing cycles in Read1, i7 index, i5 index, and Read2 (comma-delimnited with no required pattern) - TODO - If we're saying it's comma delimited there IS a required pattern, and it should be validated.

| constraint | value |
| --- | --- |
| required | `True` |

### `sequencing_read_percent_q30`
Percent of bases with Quality scores above Q30

| constraint | value |
| --- | --- |
| type | `number` |
| required | `True` |
| minimum | `0` |
| maximum | `100` |

### `sequencing_reagent_kit`
Reagent kit used for sequencing. NovaSeq6000 for example

| constraint | value |
| --- | --- |
| required | `True` |

### `transposition_kit_number`
If Tn5 came from a kit, provide the catalog number. Leave blank if not applicable.

| constraint | value |
| --- | --- |
| required | `False` |

### `transposition_method`
Modality of capturing accessible chromatin molecules. TODO - is this an enum?

| constraint | value |
| --- | --- |
| required | `True` |

### `transposition_transposase_source`
The source of the Tn5 transposase and transposon used for capturing accessible chromatin.

| constraint | value |
| --- | --- |
| required | `True` |

## Paths

### `metadata_path`
Relative path to file or directory with free-form or instrument/lab specific metadata. Optional. Leave blank if not applicable.

| constraint | value |
| --- | --- |
| required | `False` |

### `data_path`
Relative path to file or directory with instrument data. Downstream processing will depend on filename extension conventions.

| constraint | value |
| --- | --- |
| required | `True` |
