---
title: Visium (with probes)
schema_name: visium-with-probes
category: Spatial Transcriptomics
all_versions_deprecated: False
exclude_from_index: False
layout: default

---
Prepare your metadata based on the latest metadata schema using one of the template files below. See the instructions in the [Metadata Validation Workflow](https://docs.google.com/document/d/1lfgiDGbyO4K4Hz1FMsJjmJd9RdwjShtJqFYNwKpbcZY) document for more information on preparing and validating your metadata.tsv file prior to submission.

Related files:


- [üìù Excel template](https://raw.githubusercontent.com/hubmapconsortium/dataset-metadata-spreadsheet/main/visium-with-probes/latest/visium-with-probes.xlsx): For metadata entry.
- [üìù TSV template](https://raw.githubusercontent.com/hubmapconsortium/dataset-metadata-spreadsheet/main/visium-with-probes/latest/visium-with-probes.tsv): Alternative for metadata entry.


REQUIRED - For this assay, you must also prepare and submit two additional metadata.tsv files following the metadata schemas linked here for [RNAseq (with probes)](https://hubmapconsortium.github.io/ingest-validation-tools/rnaseq-with-probes/current/) and [Histology](https://hubmapconsortium.github.io/ingest-validation-tools/histology/current/). [This link](https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0) lists the set of fields that are required in the OME TIFF file XML header.

## Metadata schema


<summary><a href="https://openview.metadatacenter.org/templates/https:%2F%2Frepo.metadatacenter.org%2Ftemplates%2F4646ec9d-f3c9-4619-bc45-7e14748bb976"><b>Version 2 (use this one)</b></a></summary>



<br>

## Directory schemas
<summary><b>Version 2.1 (use this one)</b></summary>

| pattern | required? | description | dependent on |
| --- | --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset |  |
| <code>extras\/microscope_hardware\.json</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |  |
| <code>extras\/microscope_settings\.json</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |  |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |  |
| <code>raw\/[^\/]+\.gpr</code> | ‚úì | This is a 10X Genomics layout file that's generated by 10X and individualized for each Visium slide. This is a text file and can be generated using this 10X web form <https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/using/slidefile-download> along with the unique 10X Visium slide ID. |  |
| <code>raw\/additional_panels_used\.csv</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |  |
| <code>raw\/custom_probe_set\.csv</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |  |
| <code>raw\/fastq\/.*</code> | ‚úì | Raw sequencing files for the experiment |  |
| <code>raw\/fastq\/oligo\/</code> | ‚úì | Directory containing fastq files pertaining to oligo sequencing. |  |
| <code>raw\/fastq\/oligo\/[^\/]+\.fastq\.gz</code> | ‚úì | This is a gzip version of the fastq file. This file contains the cell barcode and unique molecular identifier (technical). |  |
| <code>raw\/images\/.*</code> | ‚úì | Directory containing raw image files. This directory should include at least one raw file. |  |
| <code>raw\/images\/[^\/]+_tissue\.(?:tif&#124;tiff)</code> |  | Raw microscope file for the experiment. For 10X Visium CytAssist, this would be the high resolution image produced. |  |
| <code>raw\/images\/[^\/]+_fiducial\.(?:tif&#124;tiff)</code> | ‚úì | This is the low resolution image from the 10X CytAssist instrument that includes the fiduciary markings. |  |
| <code>raw\/images\/[^\/]+\.ndpi</code> |  | Raw microscope file for the experiment |  |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |  |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |  |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |  |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |  |
| <code>lab_processed\/images\/[^\/]+\.json</code> |  | This file is the output from LoupeBrowser, when a data provider manually denotes which spots on the slide contain tissue. This file is optionally used by 10X SpaceRanger. |  |
| <code>lab_processed\/annotations\/.*</code> |  | Directory containing segmentation masks. |  |
| <code>lab_processed\/annotations\/[^\/]+\.segmentations\.ome\.tiff</code> |  | The segmentation masks should be stored as multi-channel pyramidal OME TIFF bitmasks with one channel per mask, where a single mask contains all instances of a type of object (e.g., all cells, a class of FTUs, etc). The class of objects contained in the mask is documented in the segmentation-masks.csv file. Each individual object in a mask should be represented by a unique integer pixel value starting at 1, with 0 meaning background (e.g., all pixels belonging to the first instance of a T-cell have a value of 1, the pixels for the second instance of a T-cell have a value of 2, etc). The pixel values should be unique within a mask. FTUs and other structural elements should be captured the same way as cells with segmentation masks and the appropriate channel feature definitions. | lab_processed\/annotations\/.* |
| <code>lab_processed\/annotations\/segmentation-masks\.csv</code> |  | This file contains details about each mask, with one row per mask. Each column in this file contains details describing the mask (e.g., channel number, mask name, ontological ID, etc). Each mask is stored as a channel in the segmentations.ome.tiff file and the mask name should be ontologically based and linked to the ASCT+B table where possible. The number of rows in this file should equal the number of channels in the segmentations.ome.tiff. For example, one row in this file would ontologically describe cells, if the segmentations.ome.tiff file contained a mask of all cells. A minimum set of fields (required and optional) is included below. If multiple segmentations.ome.tiff files are used, this segmentation-masks.csv file should document the masks across all of the OME TIFF files. | lab_processed\/annotations\/.* |
| <code>lab_processed\/annotations\/[^\/]+-objects\.csv</code> |  | This is a matrix where each row describes an individual object (e.g., one row per cell in the case where a mask contains all cells) and columns are features (i.e., object type, marker intensity, classification strategies, etc). One file should be created per mask with the name of the mask prepended to the file name. For example, if there‚Äôs a cell segmentation map called ‚Äúcells‚Äù then you would include a file called ‚Äúcells-objects.csv‚Äù and that file would contain one row per cell in the ‚Äúcells‚Äù mask and one column per feature, such as marker intensity and/or cell type. A minimum set of fields (required and optional) is included below. | lab_processed\/annotations\/.* |
| <code>lab_processed\/annotations\/[^\/]+\.geojson</code> |  | A GeoJSON file(s) containing the geometries of each object within a mask. For example, if the mask contains multiple FTUs, multiple cells, etc, each of the objects in the mask would be independently documented in the GeoJSON file. There would be a single GeoJSON file per mask and the name of the file should be the name of the mask. If this file is generated by QuPath, the coordinates will be in pixel units with the origin (0, 0) as the top left corner of the full-resolution image. | lab_processed\/annotations\/.* |
| <code>lab_processed\/annotations\/tissue-boundary\.geojson</code> |  | **[QA/QC]** If the boundaries of the tissue have been identified (e.g., by manual efforts), then the boundary geometry can be included as a GeoJSON file named ‚Äútissue-boundary.geojson‚Äù. | lab_processed\/annotations\/.* |
| <code>lab_processed\/annotations\/regions-of-concern\.csv</code> |  | This file and the associated GeoJSON file can be used to denote any regions in the image that may contain QA/QC concerns. For example, if there are folds in the tissue, the region of the fold can be highlighted. This file should contain one row per region and include documentation about the region and why it's being flagged. | lab_processed\/annotations\/.* |
| <code>lab_processed\/annotations\/regions-of-concern\.geojson</code> |  | This file and the associated CSV file can be used to denote any regions in the image that may contain QA/QC concerns. For example, if there are folds in the tissue, the region of the fold can be highlighted. This file should contain the geometric coordinates of each region being flagged. | lab_processed\/annotations\/.* |

<summary><b>Version 2.0</b></summary>

| pattern | required? | description | dependent on |
| --- | --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset |  |
| <code>extras\/microscope_hardware\.json</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |  |
| <code>extras\/microscope_settings\.json</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |  |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |  |
| <code>raw\/[^\/]+\.gpr</code> | ‚úì | This is a 10X Genomics layout file that's generated by 10X and individualized for each Visium slide. This is a text file and can be generated using this 10X web form <https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/using/slidefile-download> along with the unique 10X Visium slide ID. |  |
| <code>raw\/additional_panels_used\.csv</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |  |
| <code>raw\/custom_probe_set\.csv</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |  |
| <code>raw\/fastq\/.*</code> | ‚úì | Raw sequencing files for the experiment |  |
| <code>raw\/fastq\/oligo\/.*</code> | ‚úì | Directory containing fastq files pertaining to oligo sequencing. |  |
| <code>raw\/fastq\/oligo\/[^\/]+\.fastq\.gz</code> | ‚úì | This is a gzip version of the fastq file. This file contains the cell barcode and unique molecular identifier (technical). |  |
| <code>raw\/images\/.*</code> | ‚úì | Directory containing raw image files. This directory should include at least one raw file. |  |
| <code>raw\/images\/[^\/]+_tissue\.(?:tif&#124;tiff)</code> |  | Raw microscope file for the experiment. For 10X Visium CytAssist, this would be the high resolution image produced. |  |
| <code>raw\/images\/[^\/]+_fiducial\.(?:tif&#124;tiff)</code> | ‚úì | This is the low resolution image from the 10X CytAssist instrument that includes the fiduciary markings. |  |
| <code>raw\/images\/[^\/]+\.ndpi</code> |  | Raw microscope file for the experiment |  |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |  |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |  |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |  |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |  |
| <code>lab_processed\/annotations\/.*</code> |  | Directory containing segmentation masks. |  |
| <code>lab_processed\/annotations\/[^\/]+\.segmentations\.ome\.tiff</code> |  | The segmentation masks should be stored as multi-channel pyramidal OME TIFF bitmasks with one channel per mask, where a single mask contains all instances of a type of object (e.g., all cells, a class of FTUs, etc). The class of objects contained in the mask is documented in the segmentation-masks.csv file. Each individual object in a mask should be represented by a unique integer pixel value starting at 1, with 0 meaning background (e.g., all pixels belonging to the first instance of a T-cell have a value of 1, the pixels for the second instance of a T-cell have a value of 2, etc). The pixel values should be unique within a mask. FTUs and other structural elements should be captured the same way as cells with segmentation masks and the appropriate channel feature definitions. | lab_processed\/annotations\/.* |
| <code>lab_processed\/annotations\/segmentation-masks\.csv</code> |  | This file contains details about each mask, with one row per mask. Each column in this file contains details describing the mask (e.g., channel number, mask name, ontological ID, etc). Each mask is stored as a channel in the segmentations.ome.tiff file and the mask name should be ontologically based and linked to the ASCT+B table where possible. The number of rows in this file should equal the number of channels in the segmentations.ome.tiff. For example, one row in this file would ontologically describe cells, if the segmentations.ome.tiff file contained a mask of all cells. A minimum set of fields (required and optional) is included below. If multiple segmentations.ome.tiff files are used, this segmentation-masks.csv file should document the masks across all of the OME TIFF files. | lab_processed\/annotations\/.* |
| <code>lab_processed\/annotations\/[^\/]+-objects\.csv</code> |  | This is a matrix where each row describes an individual object (e.g., one row per cell in the case where a mask contains all cells) and columns are features (i.e., object type, marker intensity, classification strategies, etc). One file should be created per mask with the name of the mask prepended to the file name. For example, if there‚Äôs a cell segmentation map called ‚Äúcells‚Äù then you would include a file called ‚Äúcells-objects.csv‚Äù and that file would contain one row per cell in the ‚Äúcells‚Äù mask and one column per feature, such as marker intensity and/or cell type. A minimum set of fields (required and optional) is included below. | lab_processed\/annotations\/.* |
| <code>lab_processed\/annotations\/[^\/]+\.geojson</code> |  | A GeoJSON file(s) containing the geometries of each object within a mask. For example, if the mask contains multiple FTUs, multiple cells, etc, each of the objects in the mask would be independently documented in the GeoJSON file. There would be a single GeoJSON file per mask and the name of the file should be the name of the mask. If this file is generated by QuPath, the coordinates will be in pixel units with the origin (0, 0) as the top left corner of the full-resolution image. | lab_processed\/annotations\/.* |
| <code>lab_processed\/annotations\/tissue-boundary\.geojson</code> |  | **[QA/QC]** If the boundaries of the tissue have been identified (e.g., by manual efforts), then the boundary geometry can be included as a GeoJSON file named ‚Äútissue-boundary.geojson‚Äù. | lab_processed\/annotations\/.* |
| <code>lab_processed\/annotations\/regions-of-concern\.csv</code> |  | This file and the associated GeoJSON file can be used to denote any regions in the image that may contain QA/QC concerns. For example, if there are folds in the tissue, the region of the fold can be highlighted. This file should contain one row per region and include documentation about the region and why it's being flagged. | lab_processed\/annotations\/.* |
| <code>lab_processed\/annotations\/regions-of-concern\.geojson</code> |  | This file and the associated CSV file can be used to denote any regions in the image that may contain QA/QC concerns. For example, if there are folds in the tissue, the region of the fold can be highlighted. This file should contain the geometric coordinates of each region being flagged. | lab_processed\/annotations\/.* |

