---
title: Visium HD
schema_name: visium-hd
category: Spatial Transcriptomics
all_versions_deprecated: False
exclude_from_index: False
layout: default

---
Prepare your metadata based on the latest metadata schema using one of the template files below. See the instructions in the [Metadata Validation Workflow](https://docs.google.com/document/d/1lfgiDGbyO4K4Hz1FMsJjmJd9RdwjShtJqFYNwKpbcZY) document for more information on preparing and validating your metadata.tsv file prior to submission.

Related files:


- [üìù Excel template](https://raw.githubusercontent.com/hubmapconsortium/dataset-metadata-spreadsheet/main/visium-hd/latest/visium-hd.xlsx): For metadata entry.
- [üìù TSV template](https://raw.githubusercontent.com/hubmapconsortium/dataset-metadata-spreadsheet/main/visium-hd/latest/visium-hd.tsv): Alternative for metadata entry.


REQUIRED - For this assay, you must also prepare and submit two additional metadata.tsv files following the metadata schemas linked here for [RNAseq (with probes)](https://hubmapconsortium.github.io/ingest-validation-tools/rnaseq-with-probes/current/) and [Histology](https://hubmapconsortium.github.io/ingest-validation-tools/histology/current/). [This link](https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0) lists the set of fields that are required in the OME TIFF file XML header.

## Metadata schema


<summary><a href="https://openview.metadatacenter.org/templates/https:%2F%2Frepo.metadatacenter.org%2Ftemplates%2F3b72281f-6ed7-417d-806c-4ef781825d77"><b>Version 2 (use this one)</b></a></summary>



<br>

## Directory schemas
<summary><b>Version 2.0 (use this one)</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset |
| <code>extras\/microscope_hardware\.json$</code> | ‚úì | **[QA/QC]** A file generated by the micro-meta app that contains a description of the hardware components of the microscope. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>extras\/microscope_settings\.json$</code> |  | **[QA/QC]** A file generated by the micro-meta app that contains a description of the settings that were used to acquire the image data. Email HuBMAP Consortium Help Desk <help@hubmapconsortium.org> if help is required in generating this document. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/[^\/]+\.gpr$</code> | ‚úì | This is a 10X Genomics layout file that's generated by 10X and individualized for each Visium slide. This is a text file and can be generated using this 10X web form <https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/using/slidefile-download> along with the unique 10X Visium slide ID. |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/fastq\/.*</code> | ‚úì | Raw sequencing files for the experiment |
| <code>raw\/fastq\/oligo\/.*</code> | ‚úì | Directory containing fastq files pertaining to oligo sequencing. |
| <code>raw\/fastq\/oligo\/[^\/]+_R[^\/]+\.fastq\.gz$</code> | ‚úì | This is a gzip version of the fastq file. This file contains the cell barcode and unique molecular identifier (technical). |
| <code>raw\/images\/.*</code> | ‚úì | Directory containing raw image files. This directory should include at least one raw file. |
| <code>raw\/images\/[^\/]+_tissue\.(?:tif&#124;tiff)$</code> |  | Raw microscope file for the experiment. For 10X Visium CytAssist, this would be the high resolution image produced. |
| <code>raw\/images\/[^\/]+_fiducial\.(?:tif&#124;tiff)$</code> | ‚úì | This is the low resolution image from the 10X CytAssist instrument that includes the fiduciary markings. |
| <code>raw\/images\/[^\/]+\.ndpi$</code> |  | Raw microscope file for the experiment |
| <code>lab_processed\/.*</code> | ‚úì | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/alignment\.json$</code> | ‚úì | JSON file for the manual tissue alignment created using Loupe browser and used as input to Space Ranger. |
| <code>lab_processed\/images\/.*</code> | ‚úì | Processed image files |
| <code>lab_processed\/images\/[^\/]+\.ome\.tiff$</code> (example: <code>lab_processed/images/HBM892.MDXS.293.ome.tiff</code>) | ‚úì | OME-TIFF files (multichannel, multi-layered) produced by the microscopy experiment. If compressed, must use loss-less compression algorithm. For Visium this stitched file should only include the single capture area relevant to the current dataset. For GeoMx there will be one OME TIFF file per slide, with each slide including multiple AOIs. See the following link for the set of fields that are required in the OME TIFF file XML header. <https://docs.google.com/spreadsheets/d/1YnmdTAA0Z9MKN3OjR3Sca8pz-LNQll91wdQoRPSP6Q4/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*ome-tiff\.channels\.csv</code> | ‚úì | This file provides essential documentation pertaining to each channel of the accommpanying OME TIFF. The file should contain one row per OME TIFF channel. The required fields are detailed <https://docs.google.com/spreadsheets/d/1xEJSb0xn5C5fB3k62pj1CyHNybpt4-YtvUs5SUMS44o/edit#gid=0> |
| <code>lab_processed\/images\/[^\/]*\.tissue-boundary\.geojson$</code> |  | **[QA/QC]** If the boundaries of the tissue have been identified (e.g., by manual efforts), then the boundary geometry can be included as a GeoJSON file named ‚Äú*.tissue-boundary.geojson‚Äù. |
| <code>lab_processed\/transformations\/.*</code> |  | This directory contains transformation matrices that capture how each modality is aligned with the other and can be used to visualize overlays of multimodal data. This is needed to overlay images from the exact same tissue section (e.g., MALDI imaging mass spec, autofluorescence microscopy, MxIF, histological stains). In these cases data type may have different pixel sizes and slightly different orientations (i.e., one may be rotated relative to another). |
| <code>lab_processed\/transformations\/[^\/]+\.txt$</code> |  | Transformation matrices used to overlay images from the exact same tissue section (e.g., MALDI imaging mass spec, autofluorescence microscopy, MxIF, histological stains). |
| <code>lab_processed\/transformations\/.*</code> |  | This directory contains transformation matrices that capture how each modality is aligned with the other and can be used to visualize overlays of multimodal data. This is needed to overlay images from the exact same tissue section (e.g., MALDI imaging mass spec, autofluorescence microscopy, MxIF, histological stains). In these cases data type may have different pixel sizes and slightly different orientations (i.e., one may be rotated relative to another). |
| <code>lab_processed\/spaceranger\/.*</code> | ‚úì | Output from the 10X Genomics SpaceRanger pipeline |
| <code>lab_processed\/spaceranger\/filtered_feature_bc_matrix\.h5$</code> | ‚úì | HDF5 file containing called cell x gene matrix |
| <code>lab_processed\/spaceranger\/raw_feature_bc_matrix\.h5$</code> | ‚úì | HDF5 file containing unfiltered cell x gene matrix |
| <code>lab_processed\/spaceranger\/spatial\/.*</code> | ‚úì | Dataset level spatial files from the 10X Genomics SpaceRanger pipeline. |
| <code>lab_processed\/spaceranger\/spatial\/tissue_hires_image\.png$</code> | ‚úì | PNG image of aligned tissue at full/original resolution |
| <code>lab_processed\/spaceranger\/spatial\/tissue_lowres_image\.png$</code> | ‚úì | PNG image of aligned tissue at downscaled resolution |
| <code>lab_processed\/spaceranger\/spatial\/cytassist_image\.tiff$</code> | ‚úì | Copy of the original CytAssist image input, which is a eosin-stained brightfield image in TIFF format that contains the fiducial frame. |
| <code>lab_processed\/spaceranger\/spatial\/aligned_fiducials\.jpg$</code> | ‚úì | File used to verify that fiducial alignment was successful. Fiducial spots detected by the fiducial alignment algorithm are highlighted in red. |
| <code>lab_processed\/spaceranger\/spatial\/aligned_tissue_image\.jpg$</code> | ‚úì | Image capturing the results of the image registration by merging the registered images in a checkerboard pattern with adjacent squares alternating between the two images. |
| <code>lab_processed\/spaceranger\/spatial\/detected_tissue_image\.jpg$</code> | ‚úì | This image has either the dimensions of the CytAssist image or of the tissue_hires_image.png, and shows the following: (1) Aligned fiducial spots as hollow circles which are red (Visium HD) or blue (Visium v1/v2); (2) A blue bounding box. The interior designates the portion of the image where tissue is discriminated from background; (3) Spots/squares found under tissue are blue (Visium HD) or red (Visium v1/v2); (4) Spots/squares outside of tissue are gray. |
| <code>lab_processed\/spaceranger\/binned_outputs\/.*</code> | ‚úì | Binned outputs from the 10X Genomics SpaceRanger pipeline. |
| <code>lab_processed\/spaceranger\/binned_outputs\/square_[^\/]+\/.*</code> | ‚úì | Square-specific binned outputs from the 10X Genomics SpaceRanger pipeline. |
| <code>lab_processed\/spaceranger\/binned_outputs\/square_[^\/]+\/filtered_feature_bc_matrix\.h5$</code> | ‚úì | HDF5 file containing binned called cell x gene matrix |
| <code>lab_processed\/spaceranger\/binned_outputs\/square_[^\/]+\/raw_feature_bc_matrix\.h5$</code> | ‚úì | HDF5 file containing unfiltered binned cell x gene matrix |
| <code>lab_processed\/spaceranger\/binned_outputs\/square_[^\/]+\/spatial\/.*</code> | ‚úì | Square-specific spatial outputs from the 10X Genomics SpaceRanger pipeline |
| <code>lab_processed\/spaceranger\/binned_outputs\/square_[^\/]+\/spatial\/aligned_fiducials\.jpg$</code> | ‚úì | Square-specific file used to verify that fiducial alignment was successful. Fiducial spots detected by the fiducial alignment algorithm are highlighted in red. |
| <code>lab_processed\/spaceranger\/binned_outputs\/square_[^\/]+\/spatial\/aligned_tissue_image\.jpg$</code> | ‚úì | Square-specific image capturing the results of the image registration by merging the registered images in a checkerboard pattern with adjacent squares alternating between the two images. |
| <code>lab_processed\/spaceranger\/binned_outputs\/square_[^\/]+\/spatial\/detected_tissue_image\.jpg$</code> | ‚úì | Square-specific image displaying the following: (1) Aligned fiducial spots as hollow circles which are red (Visium HD) or blue (Visium v1/v2); (2) A blue bounding box. The interior designates the portion of the image where tissue is discriminated from background; (3) Spots/squares found under tissue are blue (Visium HD) or red (Visium v1/v2); (4) Spots/squares outside of tissue are gray. |
| <code>lab_processed\/spaceranger\/binned_outputs\/square_[^\/]+\/spatial\/tissue_positions\.parquet$</code> | ‚úì | File ontains a table with rows that correspond to spots with columns corresponding to the following field: barcode, in_tissue, array_row, array_col, pxl_row_in_fullres and pxl_col_in_fullres. |
| <code>lab_processed\/spaceranger\/binned_outputs\/square_[^\/]+\/spatial\/scalefactors_json\.json$</code> | ‚úì | Spaceranger downscales images to 2000px on the smallest side.  This file contains the spot size (diameter in pixels) at full and downscaled resolutions. |

