---
title: RNAseq (with probes)
schema_name: rnaseq-with-probes
category: Sequence Assays
all_versions_deprecated: False
exclude_from_index: False
layout: default

---
Prepare your metadata based on the latest metadata schema using one of the template files below. See the instructions in the [Metadata Validation Workflow](https://docs.google.com/document/d/1lfgiDGbyO4K4Hz1FMsJjmJd9RdwjShtJqFYNwKpbcZY) document for more information on preparing and validating your metadata.tsv file prior to submission.

Related files:


- [üìù Excel template](https://raw.githubusercontent.com/hubmapconsortium/dataset-metadata-spreadsheet/main/rnaseq-with-probes/latest/rnaseq-with-probes.xlsx): For metadata entry.
- [üìù TSV template](https://raw.githubusercontent.com/hubmapconsortium/dataset-metadata-spreadsheet/main/rnaseq-with-probes/latest/rnaseq-with-probes.tsv): Alternative for metadata entry.


For additional documentation on this dataset type, please visit [here](https://docs.google.com/document/d/1cVX_uMA5ehz3TBjrlXSb9KkRo8_5kcFUFhJaWeW9JyU).

## Metadata schema


<summary><a href="https://openview.metadatacenter.org/templates/https:%2F%2Frepo.metadatacenter.org%2Ftemplates%2Fe4df583f-95df-4113-92dc-6e9b90124d9f"><b>Version 2 (use this one)</b></a></summary>



<br>

## Directory schemas
<summary><b>Version 2.2 (use this one)</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/expected_cell_count\.txt$</code> |  | The expected cell count for the RNA sequencing dataset. This is an optional file that, if present, will be used by the HIVE's RNA sequencing analysis pipeline. With some datasets, knowing the expected cell count has improved the output of the HIVE analysis pipeline. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/fastq\/.*</code> | ‚úì | Raw sequencing files for the experiment. |
| <code>raw\/fastq\/oligo\/.*</code> | ‚úì | Directory containing fastq files pertaining to oligo sequencing. |
| <code>raw\/fastq\/oligo\/[^\/]+_R[^\/]+\.fastq\.gz$</code> | ‚úì | This is a gzip version of the fastq file. This file contains the cell barcode and unique molecular identifier (technical). |
| <code>lab_processed\/.*</code> |  | Experiment files that were processed by the lab generating the data. |
| <code>lab_processed\/cellranger\/.*</code> |  | Directory containing Cell Ranger files. |
| <code>lab_processed\/cellranger\/analysis\.tar\.gz$</code> |  | Compressed archive containing output files or data generated from a 10x Genomics Flex analysis pipeline |
| <code>lab_processed\/cellranger\/_cmdline$</code> |  | The actual CLI invocation of spaceranger command. |
| <code>lab_processed\/cellranger\/_invocation$</code> |  | Martian runtime specification determined from CLI invocation |
| <code>lab_processed\/cellranger\/_sitecheck$</code> |  | This contains details about the local environment within which SpaceRanger was run. |
| <code>lab_processed\/cellranger\/_versions$</code> |  | This contains the version number of both SpaceRanger and Martian that were used |
| <code>lab_processed\/cellranger\/web_summary\.html$</code> |  | Interactive HTML file generated by the Cell Ranger pipeline, specifically designed for analyzing Flex Gene Expression and Antibody Capture data. |
| <code>lab_processed\/cellranger\/metrics_summary\.csv$</code> |  | Contains key information about the barcoding, sequencing, and analysis process, including metrics specific to gene expression and protein detection. |
| <code>lab_processed\/cellranger\/probe_set\.csv$</code> |  | Contains the sequences, gene IDs, and other metadata for the organism-specific probes used in the Flex assay, which are necessary for the software to align sequencing reads, map them to their target genes, and count unique molecules for gene expression analysis. |
| <code>lab_processed\/cellranger\/sample_alignments\.bam$</code> |  | Contains sequencing reads that have been assigned to specific samples. |
| <code>lab_processed\/cellranger\/sample_alignments\.bam\.bai$</code> |  | Index file for the sample_alignments.bam file, which contains aligned sequencing reads from a 10x Genomics Flex experiment. |
| <code>lab_processed\/cellranger\/sample_cloupe\.cloupe$</code> |  | Proprietary output of the 10x Genomics Cell Ranger analysis pipeline for single-cell gene expression (GEX) data generated with the 10x Flex protocol. It is designed for interactive visualization and analysis in the company's dedicated Loupe Browser software. |
| <code>lab_processed\/cellranger\/sample_filtered_barcodes\.csv$</code> |  | Output of the demultiplexing process for a multiplexed Flex experiment. This file lists the cell-associated barcodes assigned to a specific sample. |
| <code>lab_processed\/cellranger\/sample_filtered_feature_bc_matrix\.h5$</code> |  | Compressed, binary file containing the filtered gene expression counts for a specific sample, suitable for analysis by single-cell RNA sequencing tools. It stores a matrix where rows represent genes (features) and columns represent cell barcodes, detailing the UMI (or ligation event) count for each feature in each filtered cell, with the data compressed in a more efficient Hierarchical Data Format (HDF5). |
| <code>lab_processed\/cellranger\/sample_filtered_feature_bc_matrix\.tar\.gz$</code> |  | Compressed archive from 10x Genomics' Cell Ranger software that contains the analyzed and processed gene expression count data for a specific sample from a Flex assay. |
| <code>lab_processed\/cellranger\/sample_molecule_info\.h5$</code> |  | Hierarchical Data Format (HDF5) file containing per-molecule information from the 10x Genomics Gene Expression Flex workflow. It stores detailed data on individual molecules detected, including their assigned genes or proteins, bar codes, and unique molecular identifiers (UMIs), along with metadata about the experiment's libraries and feature sets. |
| <code>lab_processed\/cellranger\/sample_raw_feature_bc_matrix\.h5$</code> |  | Hierarchical Data Format (HDF5) file containing the unfiltered UMI count matrix of all identified feature barcodes (genes, antibodies, etc.) versus all observed cell barcodes in a sample. It includes counts for both cell-associated and background barcodes, distinguishing it from the filtered version, and serves as a raw input for various downstream analyses by providing the complete, unadjusted data for a given sample |
| <code>lab_processed\/cellranger\/sample_raw_probe_bc_matrix\.h5$</code> |  | Raw data output from the 10x Genomics Cell Ranger pipeline for Gene Expression Flex (Flex) or Fixed RNA Profiling (FRP) assays, containing gene-level counts derived from barcoded probes targeting the whole transcriptome in formaldehyde-fixed samples. This HDF5 file stores the raw expression levels, including counts for each gene in each cell, and serves as a primary input for downstream analysis, providing a comprehensive view of gene expression in the fixed samples. |

<summary><b>Version 2.1</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/expected_cell_count\.txt$</code> |  | The expected cell count for the RNA sequencing dataset. This is an optional file that, if present, will be used by the HIVE's RNA sequencing analysis pipeline. With some datasets, knowing the expected cell count has improved the output of the HIVE analysis pipeline. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/custom_probe_set\.csv$</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/additional_panels_used\.csv$</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/fastq\/.*</code> | ‚úì | Raw sequencing files for the experiment. |
| <code>raw\/fastq\/oligo\/.*</code> | ‚úì | Directory containing fastq files pertaining to oligo sequencing. |
| <code>raw\/fastq\/oligo\/[^\/]+_R[^\/]+\.fastq\.gz$</code> | ‚úì | This is a gzip version of the fastq file. This file contains the cell barcode and unique molecular identifier (technical). |
| <code>lab_processed\/.*</code> |  | Experiment files that were processed by the lab generating the data. |

<summary><b>Version 2.0</b></summary>

| pattern | required? | description |
| --- | --- | --- |
| <code>extras\/.*</code> | ‚úì | Folder for general lab-specific files related to the dataset. |
| <code>extras\/expected_cell_count\.txt</code> |  | The expected cell count for the RNA sequencing dataset. This is an optional file that, if present, will be used by the HIVE's RNA sequencing analysis pipeline. With some datasets, knowing the expected cell count has improved the output of the HIVE analysis pipeline. |
| <code>raw\/.*</code> | ‚úì | All raw data files for the experiment. |
| <code>raw\/custom_probe_set\.csv</code> |  | This file should contain any custom probes used and must be included if the metadata field "is_custom_probes_used" is "Yes". The file should minimally include:target gene id, probe seq, probe id. The contents of this file are modeled after the 10x Genomics probe set file (see <https://support.10xgenomics.com/spatial-gene-expression-ffpe/probe-sets/probe-set-file-descriptions/probe-set-file-descriptions#probe_set_csv_file>). |
| <code>raw\/additional_panels_used\.csv</code> |  | If multiple commercial probe panels were used, then the primary probe panel should be selected in the "oligo_probe_panel" metadata field. The additional panels must be included in this file. Each panel record should include:manufacturer, model/name, product code. |
| <code>raw\/fastq\/.*</code> | ‚úì | Raw sequencing files for the experiment. |
| <code>raw\/fastq\/oligo\/.*</code> | ‚úì | Directory containing fastq files pertaining to oligo sequencing. |
| <code>raw\/fastq\/oligo\/[^\/]+_R[^\/]+\.fastq\.gz</code> | ‚úì | This is a gzip version of the fastq file. This file contains the cell barcode and unique molecular identifier (technical). |
| <code>lab_processed\/.*</code> |  | Experiment files that were processed by the lab generating the data. |

