# imc3d

Related files:

- [üìù Excel template](https://raw.githubusercontent.com/hubmapconsortium/ingest-validation-tools/master/docs/imc3d/imc3d-metadata.xlsx): For metadata entry.
- [üìù TSV template](https://raw.githubusercontent.com/hubmapconsortium/ingest-validation-tools/master/docs/imc3d/imc3d-metadata.tsv): Alternative for metadata entry.
- [üíª Source code](https://github.com/hubmapconsortium/ingest-validation-tools/edit/master/src/ingest_validation_tools/table-schemas/level-2/imc3d.yaml): Make a PR to update this doc.

## Table of contents
<details><summary>Provenance</summary>

[`donor_id`](#donor_id)<br>
[`tissue_id`](#tissue_id)<br>
</details>

<details><summary>Level 1</summary>

[`execution_datetime`](#execution_datetime)<br>
[`protocols_io_doi`](#protocols_io_doi)<br>
[`operator`](#operator)<br>
[`operator_email`](#operator_email)<br>
[`pi`](#pi)<br>
[`pi_email`](#pi_email)<br>
[`assay_category`](#assay_category)<br>
[`assay_type`](#assay_type)<br>
[`analyte_class`](#analyte_class)<br>
[`is_targeted`](#is_targeted)<br>
[`contributors_path`](#contributors_path)<br>
[`data_path`](#data_path)<br></details>

## Directory structure

| pattern | required? | description |
| --- | --- | --- |
| `mcd/[^/]+_HuBMAP_[^/]+organ[^/]+_slide[^/]+\.zip` | ‚úì | csv containing labels for sections as well as whether or not they were included in the 3D model |
| `mcd/section_report.csv` | ‚úì | **[QA/QC]** csv containing labels for sections as well as whether or not they were included in the 3D model |
| `mcd/channelnames_report.csv` | ‚úì | **[QA/QC]** CSV containing antibodies used and whether they were detected sufficiently or not |
| `3D_image_stack.ome.tiff` | ‚úì | OME.tiff file comprising all slices and channels |
| `extras` |  | Folder containing optional metadata |
| `SingleCellData/cells.csv` | ‚úì | Contains one csv file per tissue with marker intensities (untransformed, range normalized to 99th percentile), phenograph cluster label and cell type label per single cell |
| `SingleCellData/cellsinfo.txt` |  | Text file containing formatting information about cells*organ*.csv |
| `mapping/cluster_labels_image.tif` | ‚úì | Cell images labeled by cell type |
| `processed/umap_phenograph.pdf` | ‚úì | tSNE phenograph |
| `processed/CellTypeComposition_perTissue.pdf` | ‚úì | Cell type composition bar graph per tissue |
| `processed/Densityplots_perMarker.pdf` | ‚úì | **[QA/QC]** Density plots of marker intensity, separated by marker |
| `processed/celltypes.pdf` | ‚úì | Heatmap of marker expression per cluster, annotated by assigned cell type |
| `extras/.*` |  | Free-form descriptive information supplied by the TMC |
| `extras/thumbnail\.(png\|jpg)` |  | Optional thumbnail image which may be shown in search interface |

## Provenance

### `donor_id`
HuBMAP Display ID of the donor of the assayed tissue. Example: `ABC123`.

| constraint | value |
| --- | --- |
| pattern (regular expression) | `[A-Z]+[0-9]+` |
| required | `True` |

### `tissue_id`
HuBMAP Display ID of the assayed tissue. Example: `ABC123-BL-1-2-3_456`.

| constraint | value |
| --- | --- |
| pattern (regular expression) | `([A-Z]+[0-9]+)-[A-Z]{2}\d*(-\d+)+(_\d+)?` |
| required | `True` |

## Level 1

### `execution_datetime`
Start date and time of assay, typically a date-time stamped folder generated by the acquisition instrument. YYYY-MM-DD hh:mm, where YYYY is the year, MM is the month with leading 0s, and DD is the day with leading 0s, hh is the hour with leading zeros, mm are the minutes with leading zeros.

| constraint | value |
| --- | --- |
| type | `datetime` |
| format | `%Y-%m-%d %H:%M` |
| required | `True` |

### `protocols_io_doi`
DOI for protocols.io referring to the protocol for this assay.

| constraint | value |
| --- | --- |
| required | `True` |
| pattern (regular expression) | `10\.17504/.*` |
| url | prefix: `https://dx.doi.org/` |

### `operator`
Name of the person responsible for executing the assay.

| constraint | value |
| --- | --- |
| required | `True` |

### `operator_email`
Email address for the operator.

| constraint | value |
| --- | --- |
| format | `email` |
| required | `True` |

### `pi`
Name of the principal investigator responsible for the data.

| constraint | value |
| --- | --- |
| required | `True` |

### `pi_email`
Email address for the principal investigator.

| constraint | value |
| --- | --- |
| format | `email` |
| required | `True` |

### `assay_category`
Each assay is placed into one of the following 3 general categories: generation of images of microscopic entities, identification & quantitation of molecules by mass spectrometry, and determination of nucleotide sequence.

| constraint | value |
| --- | --- |
| enum | `imaging`, `mass_spectrometry`, `mass_spectrometry_imaging`, or `sequence` |
| required | `True` |

### `assay_type`
The specific type of assay being executed.

| constraint | value |
| --- | --- |
| enum | `scRNA-Seq (10xGenomics)`, `AF`, `bulk RNA`, `bulkATACseq`, `Cell DIVE`, `CODEX`, `Imaging Mass Cytometry`, `LC-MS (metabolomics)`, `LC-MS/MS (label-free proteomics)`, `Light Sheet`, `MxIF`, `MALDI-IMS`, `MS (shotgun lipidomics)`, `NanoDESI`, `NanoPOTS`, `PAS microscopy`, `scATACseq`, `sciATACseq`, `sciRNAseq`, `seqFISH`, `SNARE-seq2`, `snATACseq`, `snRNA`, `SPLiT-Seq`, `TMT (proteomics)`, `WGS`, `SNARE2-RNAseq`, `snRNAseq`, `scRNAseq-10xGenomics`, `scRNAseq`, or `Slide-seq` |
| required | `True` |

### `analyte_class`
Analytes are the target molecules being measured with the assay.

| constraint | value |
| --- | --- |
| enum | `DNA`, `RNA`, `protein`, `lipids`, `metabolites`, `polysaccharides`, or `metabolites_and_lipids` |
| required | `True` |

### `is_targeted`
Specifies whether or not a specific molecule(s) is/are targeted for detection/measurement by the assay. The CODEX analyte is protein.

| constraint | value |
| --- | --- |
| type | `boolean` |
| required | `True` |

### `contributors_path`
Relative path to file with ORCID IDs for contributors for this dataset.

| constraint | value |
| --- | --- |
| required | `True` |

### `data_path`
Relative path to file or directory with instrument data. Downstream processing will depend on filename extension conventions.

| constraint | value |
| --- | --- |
| required | `True` |
