# wgs

Related files:
- [üî¨ Background doc](TODO): More details about this type.
- [üìù TSV template](https://raw.githubusercontent.com/hubmapconsortium/ingest-validation-tools/master/docs/wgs/template.tsv): Use this to submit metadata.
- [üíª Source code](https://github.com/hubmapconsortium/ingest-validation-tools/edit/master/src/ingest_validation_tools/table-schemas/level-2/wgs.yaml): Make a PR if this doc should be updated.

## Table of contents
<details><summary>Provenance</summary>

[`donor_id`](#donor_id)<br>
[`tissue_id`](#tissue_id)<br>
</details>

<details><summary>Level 1</summary>

[`execution_datetime`](#execution_datetime)<br>
[`protocols_io_doi`](#protocols_io_doi)<br>
[`operator`](#operator)<br>
[`operator_email`](#operator_email)<br>
[`pi`](#pi)<br>
[`pi_email`](#pi_email)<br>
[`assay_category`](#assay_category)<br>
[`assay_type`](#assay_type)<br>
[`analyte_class`](#analyte_class)<br>
[`is_targeted`](#is_targeted)<br>
</details>

<details><summary>Level 2</summary>

[`acquisition_instrument_vendor`](#acquisition_instrument_vendor)<br>
[`acquisition_instrument_model`](#acquisition_instrument_model)<br>
[`gdna_fragmentation_quality_assurance`](#gdna_fragmentation_quality_assurance)<br>
[`dna_assay_input`](#dna_assay_input)<br>
[`dna_assay_input_unit`](#dna_assay_input_unit)<br>
[`library_construction_method`](#library_construction_method)<br>
[`library_construction_protocols_io_doi`](#library_construction_protocols_io_doi)<br>
[`library_layout`](#library_layout)<br>
[`library_adapter_sequence`](#library_adapter_sequence)<br>
[`library_final_yield`](#library_final_yield)<br>
[`library_final_yield_unit`](#library_final_yield_unit)<br>
[`library_average_fragment_size`](#library_average_fragment_size)<br>
[`sequencing_reagent_kit`](#sequencing_reagent_kit)<br>
[`sequencing_read_format`](#sequencing_read_format)<br>
[`sequencing_read_percent_q30`](#sequencing_read_percent_q30)<br>
[`sequencing_phix_percent`](#sequencing_phix_percent)<br>
</details>

<details><summary>Paths</summary>

[`metadata_path`](#metadata_path)<br>
[`data_path`](#data_path)<br></details>

## Provenance

### `donor_id`
HuBMAP Display ID of the donor of the assayed tissue.

| constraint | value |
| --- | --- |
| pattern | `[A-Z]+[0-9]+` |
| required | `True` |

### `tissue_id`
HuBMAP Display ID of the assayed tissue.

| constraint | value |
| --- | --- |
| pattern | `([A-Z]+[0-9]+)-(BL|BR|LB|RB|HT|LK|RK|LI|LV|LL|RL|LY\d\d|SI|SP|TH|TR|UR|OT)(-\d+)+(_\d+)?` |
| required | `True` |

## Level 1

### `execution_datetime`
Start date and time of assay, typically a date-time stamped folder generated by the acquisition instrument. YYYY-MM-DD hh:mm, where YYYY is the year, MM is the month with leading 0s, and DD is the day with leading 0s, hh is the hour with leading zeros, mm are the minutes with leading zeros.

| constraint | value |
| --- | --- |
| type | `datetime` |
| format | `%Y-%m-%d %H:%M` |
| required | `True` |

### `protocols_io_doi`
DOI for protocols.io referring to the protocol for this assay.

| constraint | value |
| --- | --- |
| required | `True` |
| pattern | `10\.17504/.*` |

### `operator`
Name of the person responsible for executing the assay.

| constraint | value |
| --- | --- |
| required | `True` |

### `operator_email`
Email address for the operator.

| constraint | value |
| --- | --- |
| format | `email` |
| required | `True` |

### `pi`
Name of the principal investigator responsible for the data.

| constraint | value |
| --- | --- |
| required | `True` |

### `pi_email`
Email address for the principal investigator.

| constraint | value |
| --- | --- |
| format | `email` |
| required | `True` |

### `assay_category`
Each assay is placed into one of the following 3 general categories: generation of images of microscopic entities, identification & quantitation of molecules by mass spectrometry, and determination of nucleotide sequence.

| constraint | value |
| --- | --- |
| enum | `imaging`, `mass_spectrometry`, or `sequence` |
| required | `True` |

### `assay_type`
The specific type of assay being executed.

| constraint | value |
| --- | --- |
| enum | `scRNA-Seq (10xGenomics)`, `AF`, `bulk RNA`, `bulkATACseq`, `CODEX`, `Imaging Mass Cytometry`, `LC-MS (metabolomics)`, `LC-MS/MS (label-free proteomics)`, `MxIF`, `IMS positive`, `IMS negative`, `MS (shotgun lipidomics)`, `PAS microscopy`, `scATACseq`, `sciATACseq`, `sciRNAseq`, `seqFISH`, `SNARE-seq2`, `snATACseq`, `snRNA`, `SPLiT-Seq`, `TMT (proteomics)`, or `WGS` |
| required | `True` |

### `analyte_class`
Analytes are the target molecules being measured with the assay.

| constraint | value |
| --- | --- |
| enum | `DNA`, `RNA`, `protein`, `lipids`, `metabolites`, or `polysaccharides` |
| required | `True` |

### `is_targeted`
Specifies whether or not a specific molecule(s) is/are targeted for detection/measurement by the assay. The CODEX analyte is protein.

| constraint | value |
| --- | --- |
| type | `boolean` |
| required | `True` |

## Level 2

### `acquisition_instrument_vendor`
An acquisition_instrument is the device that contains the signal detection hardware and signal processing software. Assays can generate signals such as light of various intensities or color or signals representing molecular mass.

| constraint | value |
| --- | --- |
| required | `True` |

### `acquisition_instrument_model`
Manufacturers of an acquisition instrument may offer various versions (models) of that instrument with different features or sensitivities. Differences in features or sensitivities may be relevant to processing or interpretation of the data.

| constraint | value |
| --- | --- |
| required | `True` |

### `gdna_fragmentation_quality_assurance`
Is the gDNA integrity good enough for WGS? This is usually checked through running a gel. TODO - Is this a boolean?

| constraint | value |
| --- | --- |
| required | `True` |

### `dna_assay_input`
Amount of DNA input into library preparation

| constraint | value |
| --- | --- |
| type | `number` |
| required | `True` |

### `dna_assay_input_unit`
Units of DNA input into library preparation

| constraint | value |
| --- | --- |
| enum | `TODO` |
| required | `True` |

### `library_construction_method`
Describes DNA library preparation kit. Modality of isolating gDNA, Fragmentation and generating sequencing libraries. TODO - enum?

| constraint | value |
| --- | --- |
| required | `True` |

### `library_construction_protocols_io_doi`
A link to the protocol document containing the library construction method (including version) that was used.

| constraint | value |
| --- | --- |
| required | `True` |
| pattern | `10\.17504/.*` |

### `library_layout`
State whether the library was generated for single-end or paired end sequencing. TODO - enum

| constraint | value |
| --- | --- |
| required | `True` |

### `library_adapter_sequence`
Adapter sequence to be used for adapter trimming. TODO - pattern

| constraint | value |
| --- | --- |
| required | `True` |

### `library_final_yield`
Total amount of library after final pcr amplification step

| constraint | value |
| --- | --- |
| type | `number` |
| required | `True` |

### `library_final_yield_unit`
Total units of library after final pcr amplification step

| constraint | value |
| --- | --- |
| enum | `ng` |
| required | `True` |

### `library_average_fragment_size`
Average size of sequencing library fragments estimated via gel electrophoresis or bioanalyzer/tapestation.

| constraint | value |
| --- | --- |
| type | `number` |
| required | `True` |

### `sequencing_reagent_kit`
Reagent kit used for sequencing

| constraint | value |
| --- | --- |
| required | `True` |

### `sequencing_read_format`
Number of sequencing cycles in Read1, i7 index, i5 index, and Read2 (R1/i7/i5/R2). TODO - pattern constraint.

| constraint | value |
| --- | --- |
| required | `True` |

### `sequencing_read_percent_q30`
Percent of bases with Quality scores above Q30

| constraint | value |
| --- | --- |
| type | `number` |
| required | `True` |
| minimum | `0` |
| maximum | `100` |

### `sequencing_phix_percent`
Percent PhiX loaded to the run

| constraint | value |
| --- | --- |
| type | `number` |
| required | `True` |
| minimum | `0` |
| maximum | `100` |

## Paths

### `metadata_path`
Relative path to file or directory with free-form or instrument/lab specific metadata. Optional. Leave blank if not applicable.

| constraint | value |
| --- | --- |
| required | `False` |

### `data_path`
Relative path to file or directory with instrument data. Downstream processing will depend on filename extension conventions.

| constraint | value |
| --- | --- |
| required | `True` |
